// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Activity.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ActivityAllOf = _interopRequireDefault(require("./ActivityAllOf"));
var _ActivityTypes = _interopRequireDefault(require("./ActivityTypes"));
var _AuthorWebhook = _interopRequireDefault(require("./AuthorWebhook"));
var _SourceWebhook = _interopRequireDefault(require("./SourceWebhook"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Activity model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Activity
 * @version 9.14.0
 */
var Activity = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Activity</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Activity
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityAllOf
   */
  function Activity() {
    _classCallCheck(this, Activity);
    _ActivityTypes["default"].initialize(this);
    _ActivityAllOf["default"].initialize(this);
    Activity.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Activity, [{
    key: "getType",
    value:
    /**
         * Returns If the author type is `user`, only `conversation:read` is supported.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Activity.TypeEnum}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets If the author type is `user`, only `conversation:read` is supported.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Activity.TypeEnum} type If the author type is `user`, only `conversation:read` is supported.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The source of the activity.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook}
         */
  }, {
    key: "getSource",
    value: function getSource() {
      return this.source;
    }

    /**
     * Sets The source of the activity.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook} source The source of the activity.
     */
  }, {
    key: "setSource",
    value: function setSource(source) {
      this['source'] = source;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook}
         */
  }, {
    key: "getAuthor",
    value: function getAuthor() {
      return this.author;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook} author
     */
  }, {
    key: "setAuthor",
    value: function setAuthor(author) {
      this['author'] = author;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Activity</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Activity} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Activity} The populated <code>Activity</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Activity();
        _ActivityTypes["default"].constructFromObject(data, obj);
        _ActivityAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], _SourceWebhook["default"]);
        }
        if (data.hasOwnProperty('author')) {
          obj['author'] = _AuthorWebhook["default"].constructFromObject(data['author']);
        }
      }
      return obj;
    }
  }]);
  return Activity;
}();
/**
 * If the author type is `user`, only `conversation:read` is supported.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Activity.TypeEnum} type
 */
Activity.prototype['type'] = undefined;

/**
 * The source of the activity.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook} source
 */
Activity.prototype['source'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook} author
 */
Activity.prototype['author'] = undefined;

// Implement ActivityTypes interface:
/**
 * If the author type is `user`, only `conversation:read` is supported.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes.TypeEnum} type
 */
_ActivityTypes["default"].prototype['type'] = undefined;
// Implement ActivityAllOf interface:
/**
 * The source of the activity.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook} source
 */
_ActivityAllOf["default"].prototype['source'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook} author
 */
_ActivityAllOf["default"].prototype['author'] = undefined;

/**
 * Allowed values for the <code>type</code> property.
 * @enum {String}
 * @readonly
 */
Activity['TypeEnum'] = {
  /**
   * value: "conversation:read"
   * @const
   */
  "conversation:read": "conversation:read",
  /**
   * value: "typing:start"
   * @const
   */
  "typing:start": "typing:start",
  /**
   * value: "typing:stop"
   * @const
   */
  "typing:stop": "typing:stop"
};
var _default = Activity;
exports["default"] = _default;
