// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _User = _interopRequireDefault(require("./User"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The AuthorWebhook model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook
 * @version 9.14.0
 */
var AuthorWebhook = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthorWebhook</code>.
   * The author of the activity.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook
   * @param type {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook.TypeEnum} The `type` of the author.
   */
  function AuthorWebhook(type) {
    _classCallCheck(this, AuthorWebhook);
    AuthorWebhook.initialize(this, type);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AuthorWebhook, [{
    key: "getType",
    value:
    /**
         * Returns The `type` of the author.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook.TypeEnum}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The `type` of the author.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook.TypeEnum} type The `type` of the author.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The id of the user. Only supported when author `type` is `user`.
         * @return {String}
         */
  }, {
    key: "getUserId",
    value: function getUserId() {
      return this.userId;
    }

    /**
     * Sets The id of the user. Only supported when author `type` is `user`.
     * @param {String} userId The id of the user. Only supported when author `type` is `user`.
     */
  }, {
    key: "setUserId",
    value: function setUserId(userId) {
      this['userId'] = userId;
    }
    /**
         * Returns The user that authored the message or activity. `profile` is included in the payload if the `includeFullUser` option is enabled.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/User}
         */
  }, {
    key: "getUser",
    value: function getUser() {
      return this.user;
    }

    /**
     * Sets The user that authored the message or activity. `profile` is included in the payload if the `includeFullUser` option is enabled.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/User} user The user that authored the message or activity. `profile` is included in the payload if the `includeFullUser` option is enabled.
     */
  }, {
    key: "setUser",
    value: function setUser(user) {
      this['user'] = user;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type) {
      obj['type'] = type;
    }

    /**
     * Constructs a <code>AuthorWebhook</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook} The populated <code>AuthorWebhook</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthorWebhook();
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('userId')) {
          obj['userId'] = _ApiClient["default"].convertToType(data['userId'], 'String');
        }
        if (data.hasOwnProperty('user')) {
          obj['user'] = _ApiClient["default"].convertToType(data['user'], _User["default"]);
        }
      }
      return obj;
    }
  }]);
  return AuthorWebhook;
}();
/**
 * The `type` of the author.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook.TypeEnum} type
 */
AuthorWebhook.prototype['type'] = undefined;

/**
 * The id of the user. Only supported when author `type` is `user`.
 * @member {String} userId
 */
AuthorWebhook.prototype['userId'] = undefined;

/**
 * The user that authored the message or activity. `profile` is included in the payload if the `includeFullUser` option is enabled.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/User} user
 */
AuthorWebhook.prototype['user'] = undefined;

/**
 * Allowed values for the <code>type</code> property.
 * @enum {String}
 * @readonly
 */
AuthorWebhook['TypeEnum'] = {
  /**
   * value: "business"
   * @const
   */
  "business": "business",
  /**
   * value: "user"
   * @const
   */
  "user": "user"
};
var _default = AuthorWebhook;
exports["default"] = _default;
