// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/CarouselMessageDisplaySettings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The CarouselMessageDisplaySettings model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/CarouselMessageDisplaySettings
 * @version 9.14.0
 */
var CarouselMessageDisplaySettings = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CarouselMessageDisplaySettings</code>.
   * Settings to adjust the carousel layout.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/CarouselMessageDisplaySettings
   */
  function CarouselMessageDisplaySettings() {
    _classCallCheck(this, CarouselMessageDisplaySettings);
    CarouselMessageDisplaySettings.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(CarouselMessageDisplaySettings, [{
    key: "getImageAspectRatio",
    value:
    /**
         * Returns Specifies how to display all carousel images. Valid values are horizontal (default) and square. Only supported in Facebook Messenger, Web Messenger, Android SDK and iOS SDK carousels.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/CarouselMessageDisplaySettings.ImageAspectRatioEnum}
         */
    function getImageAspectRatio() {
      return this.imageAspectRatio;
    }

    /**
     * Sets Specifies how to display all carousel images. Valid values are horizontal (default) and square. Only supported in Facebook Messenger, Web Messenger, Android SDK and iOS SDK carousels.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/CarouselMessageDisplaySettings.ImageAspectRatioEnum} imageAspectRatio Specifies how to display all carousel images. Valid values are horizontal (default) and square. Only supported in Facebook Messenger, Web Messenger, Android SDK and iOS SDK carousels.
     */
  }, {
    key: "setImageAspectRatio",
    value: function setImageAspectRatio(imageAspectRatio) {
      this['imageAspectRatio'] = imageAspectRatio;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>CarouselMessageDisplaySettings</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/CarouselMessageDisplaySettings} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/CarouselMessageDisplaySettings} The populated <code>CarouselMessageDisplaySettings</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CarouselMessageDisplaySettings();
        if (data.hasOwnProperty('imageAspectRatio')) {
          obj['imageAspectRatio'] = _ApiClient["default"].convertToType(data['imageAspectRatio'], 'String');
        }
      }
      return obj;
    }
  }]);
  return CarouselMessageDisplaySettings;
}();
/**
 * Specifies how to display all carousel images. Valid values are horizontal (default) and square. Only supported in Facebook Messenger, Web Messenger, Android SDK and iOS SDK carousels.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/CarouselMessageDisplaySettings.ImageAspectRatioEnum} imageAspectRatio
 */
CarouselMessageDisplaySettings.prototype['imageAspectRatio'] = undefined;

/**
 * Allowed values for the <code>imageAspectRatio</code> property.
 * @enum {String}
 * @readonly
 */
CarouselMessageDisplaySettings['ImageAspectRatioEnum'] = {
  /**
   * value: "horizontal"
   * @const
   */
  "horizontal": "horizontal",
  /**
   * value: "square"
   * @const
   */
  "square": "square"
};
var _default = CarouselMessageDisplaySettings;
exports["default"] = _default;
