// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Client.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ClientType = _interopRequireDefault(require("./ClientType"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Client model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Client
 * @version 9.14.0
 */
var Client = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Client</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Client
   */
  function Client() {
    _classCallCheck(this, Client);
    Client.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Client, [{
    key: "getId",
    value:
    /**
         * Returns The unique ID of the client.
         * @return {String}
         */
    function getId() {
      return this.id;
    }

    /**
     * Sets The unique ID of the client.
     * @param {String} id The unique ID of the client.
     */
  }, {
    key: "setId",
    value: function setId(id) {
      this['id'] = id;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientType}
         */
  }, {
    key: "getType",
    value: function getType() {
      return this.type;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientType} type
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The client status. Indicates if the client is able to receive messages or not. Can be pending, inactive, active, or blocked.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Client.StatusEnum}
         */
  }, {
    key: "getStatus",
    value: function getStatus() {
      return this.status;
    }

    /**
     * Sets The client status. Indicates if the client is able to receive messages or not. Can be pending, inactive, active, or blocked.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Client.StatusEnum} status The client status. Indicates if the client is able to receive messages or not. Can be pending, inactive, active, or blocked.
     */
  }, {
    key: "setStatus",
    value: function setStatus(status) {
      this['status'] = status;
    }
    /**
         * Returns The ID of the integration that the client was created for. Unused for clients of type sdk, as they incorporate multiple integrations.
         * @return {String}
         */
  }, {
    key: "getIntegrationId",
    value: function getIntegrationId() {
      return this.integrationId;
    }

    /**
     * Sets The ID of the integration that the client was created for. Unused for clients of type sdk, as they incorporate multiple integrations.
     * @param {String} integrationId The ID of the integration that the client was created for. Unused for clients of type sdk, as they incorporate multiple integrations.
     */
  }, {
    key: "setIntegrationId",
    value: function setIntegrationId(integrationId) {
      this['integrationId'] = integrationId;
    }
    /**
         * Returns The ID of the user on an external channel. For example, the user’s phone number for Twilio, or their page-scoped user ID for Facebook Messenger. Applies only to non-SDK clients.
         * @return {String}
         */
  }, {
    key: "getExternalId",
    value: function getExternalId() {
      return this.externalId;
    }

    /**
     * Sets The ID of the user on an external channel. For example, the user’s phone number for Twilio, or their page-scoped user ID for Facebook Messenger. Applies only to non-SDK clients.
     * @param {String} externalId The ID of the user on an external channel. For example, the user’s phone number for Twilio, or their page-scoped user ID for Facebook Messenger. Applies only to non-SDK clients.
     */
  }, {
    key: "setExternalId",
    value: function setExternalId(externalId) {
      this['externalId'] = externalId;
    }
    /**
         * Returns A datetime string with the format `YYYY-MM-DDThh:mm:ss.SSSZ` representing the last time the user interacted with this client.
         * @return {String}
         */
  }, {
    key: "getLastSeen",
    value: function getLastSeen() {
      return this.lastSeen;
    }

    /**
     * Sets A datetime string with the format `YYYY-MM-DDThh:mm:ss.SSSZ` representing the last time the user interacted with this client.
     * @param {String} lastSeen A datetime string with the format `YYYY-MM-DDThh:mm:ss.SSSZ` representing the last time the user interacted with this client.
     */
  }, {
    key: "setLastSeen",
    value: function setLastSeen(lastSeen) {
      this['lastSeen'] = lastSeen;
    }
    /**
         * Returns A timestamp signifying when the client was added to the user. Formatted as `YYYY-MM-DDThh:mm:ss.SSSZ`.
         * @return {String}
         */
  }, {
    key: "getLinkedAt",
    value: function getLinkedAt() {
      return this.linkedAt;
    }

    /**
     * Sets A timestamp signifying when the client was added to the user. Formatted as `YYYY-MM-DDThh:mm:ss.SSSZ`.
     * @param {String} linkedAt A timestamp signifying when the client was added to the user. Formatted as `YYYY-MM-DDThh:mm:ss.SSSZ`.
     */
  }, {
    key: "setLinkedAt",
    value: function setLinkedAt(linkedAt) {
      this['linkedAt'] = linkedAt;
    }
    /**
         * Returns The user's display name on the channel.
         * @return {String}
         */
  }, {
    key: "getDisplayName",
    value: function getDisplayName() {
      return this.displayName;
    }

    /**
     * Sets The user's display name on the channel.
     * @param {String} displayName The user's display name on the channel.
     */
  }, {
    key: "setDisplayName",
    value: function setDisplayName(displayName) {
      this['displayName'] = displayName;
    }
    /**
         * Returns The URL for the user's avatar on the channel.
         * @return {String}
         */
  }, {
    key: "getAvatarUrl",
    value: function getAvatarUrl() {
      return this.avatarUrl;
    }

    /**
     * Sets The URL for the user's avatar on the channel.
     * @param {String} avatarUrl The URL for the user's avatar on the channel.
     */
  }, {
    key: "setAvatarUrl",
    value: function setAvatarUrl(avatarUrl) {
      this['avatarUrl'] = avatarUrl;
    }
    /**
         * Returns A flat curated object with properties that vary for each client platform. All keys are optional and not guaranteed to be available.
         * @return {Object}
         */
  }, {
    key: "getInfo",
    value: function getInfo() {
      return this.info;
    }

    /**
     * Sets A flat curated object with properties that vary for each client platform. All keys are optional and not guaranteed to be available.
     * @param {Object} info A flat curated object with properties that vary for each client platform. All keys are optional and not guaranteed to be available.
     */
  }, {
    key: "setInfo",
    value: function setInfo(info) {
      this['info'] = info;
    }
    /**
         * Returns An object with raw properties that vary for each client platform. All keys are optional and not guaranteed to be available.
         * @return {Object}
         */
  }, {
    key: "getRaw",
    value: function getRaw() {
      return this.raw;
    }

    /**
     * Sets An object with raw properties that vary for each client platform. All keys are optional and not guaranteed to be available.
     * @param {Object} raw An object with raw properties that vary for each client platform. All keys are optional and not guaranteed to be available.
     */
  }, {
    key: "setRaw",
    value: function setRaw(raw) {
      this['raw'] = raw;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Client</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Client} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Client} The populated <code>Client</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Client();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ClientType["default"].constructFromObject(data['type']);
        }
        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
        }
        if (data.hasOwnProperty('integrationId')) {
          obj['integrationId'] = _ApiClient["default"].convertToType(data['integrationId'], 'String');
        }
        if (data.hasOwnProperty('externalId')) {
          obj['externalId'] = _ApiClient["default"].convertToType(data['externalId'], 'String');
        }
        if (data.hasOwnProperty('lastSeen')) {
          obj['lastSeen'] = _ApiClient["default"].convertToType(data['lastSeen'], 'String');
        }
        if (data.hasOwnProperty('linkedAt')) {
          obj['linkedAt'] = _ApiClient["default"].convertToType(data['linkedAt'], 'String');
        }
        if (data.hasOwnProperty('displayName')) {
          obj['displayName'] = _ApiClient["default"].convertToType(data['displayName'], 'String');
        }
        if (data.hasOwnProperty('avatarUrl')) {
          obj['avatarUrl'] = _ApiClient["default"].convertToType(data['avatarUrl'], 'String');
        }
        if (data.hasOwnProperty('info')) {
          obj['info'] = _ApiClient["default"].convertToType(data['info'], Object);
        }
        if (data.hasOwnProperty('raw')) {
          obj['raw'] = _ApiClient["default"].convertToType(data['raw'], Object);
        }
      }
      return obj;
    }
  }]);
  return Client;
}();
/**
 * The unique ID of the client.
 * @member {String} id
 */
Client.prototype['id'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientType} type
 */
Client.prototype['type'] = undefined;

/**
 * The client status. Indicates if the client is able to receive messages or not. Can be pending, inactive, active, or blocked.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Client.StatusEnum} status
 */
Client.prototype['status'] = undefined;

/**
 * The ID of the integration that the client was created for. Unused for clients of type sdk, as they incorporate multiple integrations.
 * @member {String} integrationId
 */
Client.prototype['integrationId'] = undefined;

/**
 * The ID of the user on an external channel. For example, the user’s phone number for Twilio, or their page-scoped user ID for Facebook Messenger. Applies only to non-SDK clients.
 * @member {String} externalId
 */
Client.prototype['externalId'] = undefined;

/**
 * A datetime string with the format `YYYY-MM-DDThh:mm:ss.SSSZ` representing the last time the user interacted with this client.
 * @member {String} lastSeen
 */
Client.prototype['lastSeen'] = undefined;

/**
 * A timestamp signifying when the client was added to the user. Formatted as `YYYY-MM-DDThh:mm:ss.SSSZ`.
 * @member {String} linkedAt
 */
Client.prototype['linkedAt'] = undefined;

/**
 * The user's display name on the channel.
 * @member {String} displayName
 */
Client.prototype['displayName'] = undefined;

/**
 * The URL for the user's avatar on the channel.
 * @member {String} avatarUrl
 */
Client.prototype['avatarUrl'] = undefined;

/**
 * A flat curated object with properties that vary for each client platform. All keys are optional and not guaranteed to be available.
 * @member {Object} info
 */
Client.prototype['info'] = undefined;

/**
 * An object with raw properties that vary for each client platform. All keys are optional and not guaranteed to be available.
 * @member {Object} raw
 */
Client.prototype['raw'] = undefined;

/**
 * Allowed values for the <code>status</code> property.
 * @enum {String}
 * @readonly
 */
Client['StatusEnum'] = {
  /**
   * value: "active"
   * @const
   */
  "active": "active",
  /**
   * value: "blocked"
   * @const
   */
  "blocked": "blocked",
  /**
   * value: "inactive"
   * @const
   */
  "inactive": "inactive",
  /**
   * value: "pending"
   * @const
   */
  "pending": "pending"
};
var _default = Client;
exports["default"] = _default;
