// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ClientAddEventAllOfPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Client = _interopRequireDefault(require("./Client"));
var _ConversationTruncated = _interopRequireDefault(require("./ConversationTruncated"));
var _SourceWebhook = _interopRequireDefault(require("./SourceWebhook"));
var _UserTruncated = _interopRequireDefault(require("./UserTruncated"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ClientAddEventAllOfPayload model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ClientAddEventAllOfPayload
 * @version 9.14.0
 */
var ClientAddEventAllOfPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClientAddEventAllOfPayload</code>.
   * The payload of the event. The contents of this object depend on the type of event.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ClientAddEventAllOfPayload
   */
  function ClientAddEventAllOfPayload() {
    _classCallCheck(this, ClientAddEventAllOfPayload);
    ClientAddEventAllOfPayload.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ClientAddEventAllOfPayload, [{
    key: "getConversation",
    value:
    /**
         * Returns The conversation associated with the creation of the client. This field is only present when the reason is `channelLinking` and when attaching the client to a specific conversation. 
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated}
         */
    function getConversation() {
      return this.conversation;
    }

    /**
     * Sets The conversation associated with the creation of the client. This field is only present when the reason is `channelLinking` and when attaching the client to a specific conversation. 
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} conversation The conversation associated with the creation of the client. This field is only present when the reason is `channelLinking` and when attaching the client to a specific conversation. 
     */
  }, {
    key: "setConversation",
    value: function setConversation(conversation) {
      this['conversation'] = conversation;
    }
    /**
         * Returns The user associated with the client.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated}
         */
  }, {
    key: "getUser",
    value: function getUser() {
      return this.user;
    }

    /**
     * Sets The user associated with the client.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated} user The user associated with the client.
     */
  }, {
    key: "setUser",
    value: function setUser(user) {
      this['user'] = user;
    }
    /**
         * Returns The client that was just created.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Client}
         */
  }, {
    key: "getClient",
    value: function getClient() {
      return this.client;
    }

    /**
     * Sets The client that was just created.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Client} client The client that was just created.
     */
  }, {
    key: "setClient",
    value: function setClient(client) {
      this['client'] = client;
    }
    /**
         * Returns The reason for which the client was added. * `channelLinking` - The client was created as a result of initiating a channel link. * `sdkLogin` - The client was created as a result of logging into an SDK device. * `authCode` - The client was created as a result of initializing an SDK client with an `authCode`. 
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientAddEventAllOfPayload.ReasonEnum}
         */
  }, {
    key: "getReason",
    value: function getReason() {
      return this.reason;
    }

    /**
     * Sets The reason for which the client was added. * `channelLinking` - The client was created as a result of initiating a channel link. * `sdkLogin` - The client was created as a result of logging into an SDK device. * `authCode` - The client was created as a result of initializing an SDK client with an `authCode`. 
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientAddEventAllOfPayload.ReasonEnum} reason The reason for which the client was added. * `channelLinking` - The client was created as a result of initiating a channel link. * `sdkLogin` - The client was created as a result of logging into an SDK device. * `authCode` - The client was created as a result of initializing an SDK client with an `authCode`. 
     */
  }, {
    key: "setReason",
    value: function setReason(reason) {
      this['reason'] = reason;
    }
    /**
         * Returns The source where this event originated from. This could be the API or an SDK device.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook}
         */
  }, {
    key: "getSource",
    value: function getSource() {
      return this.source;
    }

    /**
     * Sets The source where this event originated from. This could be the API or an SDK device.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook} source The source where this event originated from. This could be the API or an SDK device.
     */
  }, {
    key: "setSource",
    value: function setSource(source) {
      this['source'] = source;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ClientAddEventAllOfPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientAddEventAllOfPayload} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientAddEventAllOfPayload} The populated <code>ClientAddEventAllOfPayload</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClientAddEventAllOfPayload();
        if (data.hasOwnProperty('conversation')) {
          obj['conversation'] = _ApiClient["default"].convertToType(data['conversation'], _ConversationTruncated["default"]);
        }
        if (data.hasOwnProperty('user')) {
          obj['user'] = _ApiClient["default"].convertToType(data['user'], _UserTruncated["default"]);
        }
        if (data.hasOwnProperty('client')) {
          obj['client'] = _ApiClient["default"].convertToType(data['client'], _Client["default"]);
        }
        if (data.hasOwnProperty('reason')) {
          obj['reason'] = _ApiClient["default"].convertToType(data['reason'], 'String');
        }
        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], _SourceWebhook["default"]);
        }
      }
      return obj;
    }
  }]);
  return ClientAddEventAllOfPayload;
}();
/**
 * The conversation associated with the creation of the client. This field is only present when the reason is `channelLinking` and when attaching the client to a specific conversation. 
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} conversation
 */
ClientAddEventAllOfPayload.prototype['conversation'] = undefined;

/**
 * The user associated with the client.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated} user
 */
ClientAddEventAllOfPayload.prototype['user'] = undefined;

/**
 * The client that was just created.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Client} client
 */
ClientAddEventAllOfPayload.prototype['client'] = undefined;

/**
 * The reason for which the client was added. * `channelLinking` - The client was created as a result of initiating a channel link. * `sdkLogin` - The client was created as a result of logging into an SDK device. * `authCode` - The client was created as a result of initializing an SDK client with an `authCode`. 
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientAddEventAllOfPayload.ReasonEnum} reason
 */
ClientAddEventAllOfPayload.prototype['reason'] = undefined;

/**
 * The source where this event originated from. This could be the API or an SDK device.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook} source
 */
ClientAddEventAllOfPayload.prototype['source'] = undefined;

/**
 * Allowed values for the <code>reason</code> property.
 * @enum {String}
 * @readonly
 */
ClientAddEventAllOfPayload['ReasonEnum'] = {
  /**
   * value: "channelLinking"
   * @const
   */
  "channelLinking": "channelLinking",
  /**
   * value: "sdkLogin"
   * @const
   */
  "sdkLogin": "sdkLogin",
  /**
   * value: "authCode"
   * @const
   */
  "authCode": "authCode"
};
var _default = ClientAddEventAllOfPayload;
exports["default"] = _default;
