// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Conversation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ConversationAllOf = _interopRequireDefault(require("./ConversationAllOf"));
var _ConversationTruncated = _interopRequireDefault(require("./ConversationTruncated"));
var _ConversationType = _interopRequireDefault(require("./ConversationType"));
var _SwitchboardIntegrationWebhook = _interopRequireDefault(require("./SwitchboardIntegrationWebhook"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Conversation model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Conversation
 * @version 9.14.0
 */
var Conversation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Conversation</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Conversation
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationAllOf
   */
  function Conversation() {
    _classCallCheck(this, Conversation);
    _ConversationTruncated["default"].initialize(this);
    _ConversationAllOf["default"].initialize(this);
    Conversation.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Conversation, [{
    key: "getId",
    value:
    /**
         * Returns The unique ID of the conversation.
         * @return {String}
         */
    function getId() {
      return this.id;
    }

    /**
     * Sets The unique ID of the conversation.
     * @param {String} id The unique ID of the conversation.
     */
  }, {
    key: "setId",
    value: function setId(id) {
      this['id'] = id;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationType}
         */
  }, {
    key: "getType",
    value: function getType() {
      return this.type;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationType} type
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
         * @return {Object}
         */
  }, {
    key: "getMetadata",
    value: function getMetadata() {
      return this.metadata;
    }

    /**
     * Sets Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
     * @param {Object} metadata Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
     */
  }, {
    key: "setMetadata",
    value: function setMetadata(metadata) {
      this['metadata'] = metadata;
    }
    /**
         * Returns The current switchboard integration that is in control of the conversation. This field is omitted if no `activeSwitchboardIntegration` exists for the conversation.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook}
         */
  }, {
    key: "getActiveSwitchboardIntegration",
    value: function getActiveSwitchboardIntegration() {
      return this.activeSwitchboardIntegration;
    }

    /**
     * Sets The current switchboard integration that is in control of the conversation. This field is omitted if no `activeSwitchboardIntegration` exists for the conversation.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} activeSwitchboardIntegration The current switchboard integration that is in control of the conversation. This field is omitted if no `activeSwitchboardIntegration` exists for the conversation.
     */
  }, {
    key: "setActiveSwitchboardIntegration",
    value: function setActiveSwitchboardIntegration(activeSwitchboardIntegration) {
      this['activeSwitchboardIntegration'] = activeSwitchboardIntegration;
    }
    /**
         * Returns The switchboard integration that is awaiting control. This field is omitted if no switchboard integration has been previously offered control.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook}
         */
  }, {
    key: "getPendingSwitchboardIntegration",
    value: function getPendingSwitchboardIntegration() {
      return this.pendingSwitchboardIntegration;
    }

    /**
     * Sets The switchboard integration that is awaiting control. This field is omitted if no switchboard integration has been previously offered control.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} pendingSwitchboardIntegration The switchboard integration that is awaiting control. This field is omitted if no switchboard integration has been previously offered control.
     */
  }, {
    key: "setPendingSwitchboardIntegration",
    value: function setPendingSwitchboardIntegration(pendingSwitchboardIntegration) {
      this['pendingSwitchboardIntegration'] = pendingSwitchboardIntegration;
    }
    /**
         * Returns Whether the conversation is the default conversation for the user. Will be true for the first personal conversation created for the user, and false in all other cases. 
         * @return {Boolean}
         */
  }, {
    key: "getIsDefault",
    value: function getIsDefault() {
      return this.isDefault;
    }

    /**
     * Sets Whether the conversation is the default conversation for the user. Will be true for the first personal conversation created for the user, and false in all other cases. 
     * @param {Boolean} isDefault Whether the conversation is the default conversation for the user. Will be true for the first personal conversation created for the user, and false in all other cases. 
     */
  }, {
    key: "setIsDefault",
    value: function setIsDefault(isDefault) {
      this['isDefault'] = isDefault;
    }
    /**
         * Returns A friendly name for the conversation, may be displayed to the business or the user. 
         * @return {String}
         */
  }, {
    key: "getDisplayName",
    value: function getDisplayName() {
      return this.displayName;
    }

    /**
     * Sets A friendly name for the conversation, may be displayed to the business or the user. 
     * @param {String} displayName A friendly name for the conversation, may be displayed to the business or the user. 
     */
  }, {
    key: "setDisplayName",
    value: function setDisplayName(displayName) {
      this['displayName'] = displayName;
    }
    /**
         * Returns A short text describing the conversation.
         * @return {String}
         */
  }, {
    key: "getDescription",
    value: function getDescription() {
      return this.description;
    }

    /**
     * Sets A short text describing the conversation.
     * @param {String} description A short text describing the conversation.
     */
  }, {
    key: "setDescription",
    value: function setDescription(description) {
      this['description'] = description;
    }
    /**
         * Returns A custom conversation icon url. The image must be in either JPG, PNG, or GIF format
         * @return {String}
         */
  }, {
    key: "getIconUrl",
    value: function getIconUrl() {
      return this.iconUrl;
    }

    /**
     * Sets A custom conversation icon url. The image must be in either JPG, PNG, or GIF format
     * @param {String} iconUrl A custom conversation icon url. The image must be in either JPG, PNG, or GIF format
     */
  }, {
    key: "setIconUrl",
    value: function setIconUrl(iconUrl) {
      this['iconUrl'] = iconUrl;
    }
    /**
         * Returns A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the conversation was last marked as read with role business. 
         * @return {String}
         */
  }, {
    key: "getBusinessLastRead",
    value: function getBusinessLastRead() {
      return this.businessLastRead;
    }

    /**
     * Sets A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the conversation was last marked as read with role business. 
     * @param {String} businessLastRead A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the conversation was last marked as read with role business. 
     */
  }, {
    key: "setBusinessLastRead",
    value: function setBusinessLastRead(businessLastRead) {
      this['businessLastRead'] = businessLastRead;
    }
    /**
         * Returns A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the last message was received in the conversation, or the creation time if no messages have been received yet. 
         * @return {String}
         */
  }, {
    key: "getLastUpdatedAt",
    value: function getLastUpdatedAt() {
      return this.lastUpdatedAt;
    }

    /**
     * Sets A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the last message was received in the conversation, or the creation time if no messages have been received yet. 
     * @param {String} lastUpdatedAt A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the last message was received in the conversation, or the creation time if no messages have been received yet. 
     */
  }, {
    key: "setLastUpdatedAt",
    value: function setLastUpdatedAt(lastUpdatedAt) {
      this['lastUpdatedAt'] = lastUpdatedAt;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Conversation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Conversation} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Conversation} The populated <code>Conversation</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Conversation();
        _ConversationTruncated["default"].constructFromObject(data, obj);
        _ConversationAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ConversationType["default"].constructFromObject(data['type']);
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
        if (data.hasOwnProperty('activeSwitchboardIntegration')) {
          obj['activeSwitchboardIntegration'] = _ApiClient["default"].convertToType(data['activeSwitchboardIntegration'], _SwitchboardIntegrationWebhook["default"]);
        }
        if (data.hasOwnProperty('pendingSwitchboardIntegration')) {
          obj['pendingSwitchboardIntegration'] = _ApiClient["default"].convertToType(data['pendingSwitchboardIntegration'], _SwitchboardIntegrationWebhook["default"]);
        }
        if (data.hasOwnProperty('isDefault')) {
          obj['isDefault'] = _ApiClient["default"].convertToType(data['isDefault'], 'Boolean');
        }
        if (data.hasOwnProperty('displayName')) {
          obj['displayName'] = _ApiClient["default"].convertToType(data['displayName'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('iconUrl')) {
          obj['iconUrl'] = _ApiClient["default"].convertToType(data['iconUrl'], 'String');
        }
        if (data.hasOwnProperty('businessLastRead')) {
          obj['businessLastRead'] = _ApiClient["default"].convertToType(data['businessLastRead'], 'String');
        }
        if (data.hasOwnProperty('lastUpdatedAt')) {
          obj['lastUpdatedAt'] = _ApiClient["default"].convertToType(data['lastUpdatedAt'], 'String');
        }
      }
      return obj;
    }
  }]);
  return Conversation;
}();
/**
 * The unique ID of the conversation.
 * @member {String} id
 */
Conversation.prototype['id'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationType} type
 */
Conversation.prototype['type'] = undefined;

/**
 * Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
 * @member {Object} metadata
 */
Conversation.prototype['metadata'] = undefined;

/**
 * The current switchboard integration that is in control of the conversation. This field is omitted if no `activeSwitchboardIntegration` exists for the conversation.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} activeSwitchboardIntegration
 */
Conversation.prototype['activeSwitchboardIntegration'] = undefined;

/**
 * The switchboard integration that is awaiting control. This field is omitted if no switchboard integration has been previously offered control.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} pendingSwitchboardIntegration
 */
Conversation.prototype['pendingSwitchboardIntegration'] = undefined;

/**
 * Whether the conversation is the default conversation for the user. Will be true for the first personal conversation created for the user, and false in all other cases. 
 * @member {Boolean} isDefault
 */
Conversation.prototype['isDefault'] = undefined;

/**
 * A friendly name for the conversation, may be displayed to the business or the user. 
 * @member {String} displayName
 */
Conversation.prototype['displayName'] = undefined;

/**
 * A short text describing the conversation.
 * @member {String} description
 */
Conversation.prototype['description'] = undefined;

/**
 * A custom conversation icon url. The image must be in either JPG, PNG, or GIF format
 * @member {String} iconUrl
 */
Conversation.prototype['iconUrl'] = undefined;

/**
 * A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the conversation was last marked as read with role business. 
 * @member {String} businessLastRead
 */
Conversation.prototype['businessLastRead'] = undefined;

/**
 * A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the last message was received in the conversation, or the creation time if no messages have been received yet. 
 * @member {String} lastUpdatedAt
 */
Conversation.prototype['lastUpdatedAt'] = undefined;

// Implement ConversationTruncated interface:
/**
 * The unique ID of the conversation.
 * @member {String} id
 */
_ConversationTruncated["default"].prototype['id'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationType} type
 */
_ConversationTruncated["default"].prototype['type'] = undefined;
/**
 * Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
 * @member {Object} metadata
 */
_ConversationTruncated["default"].prototype['metadata'] = undefined;
/**
 * The current switchboard integration that is in control of the conversation. This field is omitted if no `activeSwitchboardIntegration` exists for the conversation.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} activeSwitchboardIntegration
 */
_ConversationTruncated["default"].prototype['activeSwitchboardIntegration'] = undefined;
/**
 * The switchboard integration that is awaiting control. This field is omitted if no switchboard integration has been previously offered control.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} pendingSwitchboardIntegration
 */
_ConversationTruncated["default"].prototype['pendingSwitchboardIntegration'] = undefined;
// Implement ConversationAllOf interface:
/**
 * Whether the conversation is the default conversation for the user. Will be true for the first personal conversation created for the user, and false in all other cases. 
 * @member {Boolean} isDefault
 */
_ConversationAllOf["default"].prototype['isDefault'] = undefined;
/**
 * A friendly name for the conversation, may be displayed to the business or the user. 
 * @member {String} displayName
 */
_ConversationAllOf["default"].prototype['displayName'] = undefined;
/**
 * A short text describing the conversation.
 * @member {String} description
 */
_ConversationAllOf["default"].prototype['description'] = undefined;
/**
 * A custom conversation icon url. The image must be in either JPG, PNG, or GIF format
 * @member {String} iconUrl
 */
_ConversationAllOf["default"].prototype['iconUrl'] = undefined;
/**
 * A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the conversation was last marked as read with role business. 
 * @member {String} businessLastRead
 */
_ConversationAllOf["default"].prototype['businessLastRead'] = undefined;
/**
 * A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the last message was received in the conversation, or the creation time if no messages have been received yet. 
 * @member {String} lastUpdatedAt
 */
_ConversationAllOf["default"].prototype['lastUpdatedAt'] = undefined;
var _default = Conversation;
exports["default"] = _default;
