// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ConversationType = _interopRequireDefault(require("./ConversationType"));
var _SwitchboardIntegrationWebhook = _interopRequireDefault(require("./SwitchboardIntegrationWebhook"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ConversationTruncated model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated
 * @version 9.14.0
 */
var ConversationTruncated = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationTruncated</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated
   */
  function ConversationTruncated() {
    _classCallCheck(this, ConversationTruncated);
    ConversationTruncated.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ConversationTruncated, [{
    key: "getId",
    value:
    /**
         * Returns The unique ID of the conversation.
         * @return {String}
         */
    function getId() {
      return this.id;
    }

    /**
     * Sets The unique ID of the conversation.
     * @param {String} id The unique ID of the conversation.
     */
  }, {
    key: "setId",
    value: function setId(id) {
      this['id'] = id;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationType}
         */
  }, {
    key: "getType",
    value: function getType() {
      return this.type;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationType} type
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
         * @return {Object}
         */
  }, {
    key: "getMetadata",
    value: function getMetadata() {
      return this.metadata;
    }

    /**
     * Sets Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
     * @param {Object} metadata Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
     */
  }, {
    key: "setMetadata",
    value: function setMetadata(metadata) {
      this['metadata'] = metadata;
    }
    /**
         * Returns The current switchboard integration that is in control of the conversation. This field is omitted if no `activeSwitchboardIntegration` exists for the conversation.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook}
         */
  }, {
    key: "getActiveSwitchboardIntegration",
    value: function getActiveSwitchboardIntegration() {
      return this.activeSwitchboardIntegration;
    }

    /**
     * Sets The current switchboard integration that is in control of the conversation. This field is omitted if no `activeSwitchboardIntegration` exists for the conversation.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} activeSwitchboardIntegration The current switchboard integration that is in control of the conversation. This field is omitted if no `activeSwitchboardIntegration` exists for the conversation.
     */
  }, {
    key: "setActiveSwitchboardIntegration",
    value: function setActiveSwitchboardIntegration(activeSwitchboardIntegration) {
      this['activeSwitchboardIntegration'] = activeSwitchboardIntegration;
    }
    /**
         * Returns The switchboard integration that is awaiting control. This field is omitted if no switchboard integration has been previously offered control.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook}
         */
  }, {
    key: "getPendingSwitchboardIntegration",
    value: function getPendingSwitchboardIntegration() {
      return this.pendingSwitchboardIntegration;
    }

    /**
     * Sets The switchboard integration that is awaiting control. This field is omitted if no switchboard integration has been previously offered control.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} pendingSwitchboardIntegration The switchboard integration that is awaiting control. This field is omitted if no switchboard integration has been previously offered control.
     */
  }, {
    key: "setPendingSwitchboardIntegration",
    value: function setPendingSwitchboardIntegration(pendingSwitchboardIntegration) {
      this['pendingSwitchboardIntegration'] = pendingSwitchboardIntegration;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ConversationTruncated</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} The populated <code>ConversationTruncated</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationTruncated();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ConversationType["default"].constructFromObject(data['type']);
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
        if (data.hasOwnProperty('activeSwitchboardIntegration')) {
          obj['activeSwitchboardIntegration'] = _ApiClient["default"].convertToType(data['activeSwitchboardIntegration'], _SwitchboardIntegrationWebhook["default"]);
        }
        if (data.hasOwnProperty('pendingSwitchboardIntegration')) {
          obj['pendingSwitchboardIntegration'] = _ApiClient["default"].convertToType(data['pendingSwitchboardIntegration'], _SwitchboardIntegrationWebhook["default"]);
        }
      }
      return obj;
    }
  }]);
  return ConversationTruncated;
}();
/**
 * The unique ID of the conversation.
 * @member {String} id
 */
ConversationTruncated.prototype['id'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationType} type
 */
ConversationTruncated.prototype['type'] = undefined;

/**
 * Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
 * @member {Object} metadata
 */
ConversationTruncated.prototype['metadata'] = undefined;

/**
 * The current switchboard integration that is in control of the conversation. This field is omitted if no `activeSwitchboardIntegration` exists for the conversation.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} activeSwitchboardIntegration
 */
ConversationTruncated.prototype['activeSwitchboardIntegration'] = undefined;

/**
 * The switchboard integration that is awaiting control. This field is omitted if no switchboard integration has been previously offered control.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationWebhook} pendingSwitchboardIntegration
 */
ConversationTruncated.prototype['pendingSwitchboardIntegration'] = undefined;
var _default = ConversationTruncated;
exports["default"] = _default;
