// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ConversationAllOf model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationAllOf
 * @version 9.14.0
 */
var ConversationAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationAllOf</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationAllOf
   */
  function ConversationAllOf() {
    _classCallCheck(this, ConversationAllOf);
    ConversationAllOf.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ConversationAllOf, [{
    key: "getIsDefault",
    value:
    /**
         * Returns Whether the conversation is the default conversation for the user. Will be true for the first personal conversation created for the user, and false in all other cases. 
         * @return {Boolean}
         */
    function getIsDefault() {
      return this.isDefault;
    }

    /**
     * Sets Whether the conversation is the default conversation for the user. Will be true for the first personal conversation created for the user, and false in all other cases. 
     * @param {Boolean} isDefault Whether the conversation is the default conversation for the user. Will be true for the first personal conversation created for the user, and false in all other cases. 
     */
  }, {
    key: "setIsDefault",
    value: function setIsDefault(isDefault) {
      this['isDefault'] = isDefault;
    }
    /**
         * Returns A friendly name for the conversation, may be displayed to the business or the user. 
         * @return {String}
         */
  }, {
    key: "getDisplayName",
    value: function getDisplayName() {
      return this.displayName;
    }

    /**
     * Sets A friendly name for the conversation, may be displayed to the business or the user. 
     * @param {String} displayName A friendly name for the conversation, may be displayed to the business or the user. 
     */
  }, {
    key: "setDisplayName",
    value: function setDisplayName(displayName) {
      this['displayName'] = displayName;
    }
    /**
         * Returns A short text describing the conversation.
         * @return {String}
         */
  }, {
    key: "getDescription",
    value: function getDescription() {
      return this.description;
    }

    /**
     * Sets A short text describing the conversation.
     * @param {String} description A short text describing the conversation.
     */
  }, {
    key: "setDescription",
    value: function setDescription(description) {
      this['description'] = description;
    }
    /**
         * Returns A custom conversation icon url. The image must be in either JPG, PNG, or GIF format
         * @return {String}
         */
  }, {
    key: "getIconUrl",
    value: function getIconUrl() {
      return this.iconUrl;
    }

    /**
     * Sets A custom conversation icon url. The image must be in either JPG, PNG, or GIF format
     * @param {String} iconUrl A custom conversation icon url. The image must be in either JPG, PNG, or GIF format
     */
  }, {
    key: "setIconUrl",
    value: function setIconUrl(iconUrl) {
      this['iconUrl'] = iconUrl;
    }
    /**
         * Returns A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the conversation was last marked as read with role business. 
         * @return {String}
         */
  }, {
    key: "getBusinessLastRead",
    value: function getBusinessLastRead() {
      return this.businessLastRead;
    }

    /**
     * Sets A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the conversation was last marked as read with role business. 
     * @param {String} businessLastRead A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the conversation was last marked as read with role business. 
     */
  }, {
    key: "setBusinessLastRead",
    value: function setBusinessLastRead(businessLastRead) {
      this['businessLastRead'] = businessLastRead;
    }
    /**
         * Returns A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the last message was received in the conversation, or the creation time if no messages have been received yet. 
         * @return {String}
         */
  }, {
    key: "getLastUpdatedAt",
    value: function getLastUpdatedAt() {
      return this.lastUpdatedAt;
    }

    /**
     * Sets A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the last message was received in the conversation, or the creation time if no messages have been received yet. 
     * @param {String} lastUpdatedAt A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the last message was received in the conversation, or the creation time if no messages have been received yet. 
     */
  }, {
    key: "setLastUpdatedAt",
    value: function setLastUpdatedAt(lastUpdatedAt) {
      this['lastUpdatedAt'] = lastUpdatedAt;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ConversationAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationAllOf} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationAllOf} The populated <code>ConversationAllOf</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationAllOf();
        if (data.hasOwnProperty('isDefault')) {
          obj['isDefault'] = _ApiClient["default"].convertToType(data['isDefault'], 'Boolean');
        }
        if (data.hasOwnProperty('displayName')) {
          obj['displayName'] = _ApiClient["default"].convertToType(data['displayName'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('iconUrl')) {
          obj['iconUrl'] = _ApiClient["default"].convertToType(data['iconUrl'], 'String');
        }
        if (data.hasOwnProperty('businessLastRead')) {
          obj['businessLastRead'] = _ApiClient["default"].convertToType(data['businessLastRead'], 'String');
        }
        if (data.hasOwnProperty('lastUpdatedAt')) {
          obj['lastUpdatedAt'] = _ApiClient["default"].convertToType(data['lastUpdatedAt'], 'String');
        }
      }
      return obj;
    }
  }]);
  return ConversationAllOf;
}();
/**
 * Whether the conversation is the default conversation for the user. Will be true for the first personal conversation created for the user, and false in all other cases. 
 * @member {Boolean} isDefault
 */
ConversationAllOf.prototype['isDefault'] = undefined;

/**
 * A friendly name for the conversation, may be displayed to the business or the user. 
 * @member {String} displayName
 */
ConversationAllOf.prototype['displayName'] = undefined;

/**
 * A short text describing the conversation.
 * @member {String} description
 */
ConversationAllOf.prototype['description'] = undefined;

/**
 * A custom conversation icon url. The image must be in either JPG, PNG, or GIF format
 * @member {String} iconUrl
 */
ConversationAllOf.prototype['iconUrl'] = undefined;

/**
 * A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the conversation was last marked as read with role business. 
 * @member {String} businessLastRead
 */
ConversationAllOf.prototype['businessLastRead'] = undefined;

/**
 * A datetime string with the format YYYY-MM-DDThh:mm:ss.SSSZ representing the moment the last message was received in the conversation, or the creation time if no messages have been received yet. 
 * @member {String} lastUpdatedAt
 */
ConversationAllOf.prototype['lastUpdatedAt'] = undefined;
var _default = ConversationAllOf;
exports["default"] = _default;
