// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageEventAllOfPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ConversationTruncated = _interopRequireDefault(require("./ConversationTruncated"));
var _MessageWebhook = _interopRequireDefault(require("./MessageWebhook"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ConversationMessageEventAllOfPayload model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageEventAllOfPayload
 * @version 9.14.0
 */
var ConversationMessageEventAllOfPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationMessageEventAllOfPayload</code>.
   * The payload of the event. The contents of this object depend on the type of event.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageEventAllOfPayload
   */
  function ConversationMessageEventAllOfPayload() {
    _classCallCheck(this, ConversationMessageEventAllOfPayload);
    ConversationMessageEventAllOfPayload.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ConversationMessageEventAllOfPayload, [{
    key: "getConversation",
    value:
    /**
         * Returns The conversation in which the message was sent.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated}
         */
    function getConversation() {
      return this.conversation;
    }

    /**
     * Sets The conversation in which the message was sent.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} conversation The conversation in which the message was sent.
     */
  }, {
    key: "setConversation",
    value: function setConversation(conversation) {
      this['conversation'] = conversation;
    }
    /**
         * Returns The message that was sent.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageWebhook}
         */
  }, {
    key: "getMessage",
    value: function getMessage() {
      return this.message;
    }

    /**
     * Sets The message that was sent.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageWebhook} message The message that was sent.
     */
  }, {
    key: "setMessage",
    value: function setMessage(message) {
      this['message'] = message;
    }
    /**
         * Returns Messages sent with the Notification API since the last user message.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/MessageWebhook>}
         */
  }, {
    key: "getRecentNotifications",
    value: function getRecentNotifications() {
      return this.recentNotifications;
    }

    /**
     * Sets Messages sent with the Notification API since the last user message.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/MessageWebhook>} recentNotifications Messages sent with the Notification API since the last user message.
     */
  }, {
    key: "setRecentNotifications",
    value: function setRecentNotifications(recentNotifications) {
      this['recentNotifications'] = recentNotifications;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ConversationMessageEventAllOfPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageEventAllOfPayload} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageEventAllOfPayload} The populated <code>ConversationMessageEventAllOfPayload</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationMessageEventAllOfPayload();
        if (data.hasOwnProperty('conversation')) {
          obj['conversation'] = _ApiClient["default"].convertToType(data['conversation'], _ConversationTruncated["default"]);
        }
        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], _MessageWebhook["default"]);
        }
        if (data.hasOwnProperty('recentNotifications')) {
          obj['recentNotifications'] = _ApiClient["default"].convertToType(data['recentNotifications'], [_MessageWebhook["default"]]);
        }
      }
      return obj;
    }
  }]);
  return ConversationMessageEventAllOfPayload;
}();
/**
 * The conversation in which the message was sent.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} conversation
 */
ConversationMessageEventAllOfPayload.prototype['conversation'] = undefined;

/**
 * The message that was sent.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageWebhook} message
 */
ConversationMessageEventAllOfPayload.prototype['message'] = undefined;

/**
 * Messages sent with the Notification API since the last user message.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/MessageWebhook>} recentNotifications
 */
ConversationMessageEventAllOfPayload.prototype['recentNotifications'] = undefined;
var _default = ConversationMessageEventAllOfPayload;
exports["default"] = _default;
