// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Identity.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Identity model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Identity
 * @version 9.14.0
 */
var Identity = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Identity</code>.
   * A connected user identity, such as an email.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Identity
   */
  function Identity() {
    _classCallCheck(this, Identity);
    Identity.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Identity, [{
    key: "getType",
    value:
    /**
         * Returns The type of identity.
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The type of identity.
     * @param {String} type The type of identity.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The identity value.
         * @return {String}
         */
  }, {
    key: "getValue",
    value: function getValue() {
      return this.value;
    }

    /**
     * Sets The identity value.
     * @param {String} value The identity value.
     */
  }, {
    key: "setValue",
    value: function setValue(value) {
      this['value'] = value;
    }
    /**
         * Returns The type of verification performed on the identity.
         * @return {String}
         */
  }, {
    key: "getVerification",
    value: function getVerification() {
      return this.verification;
    }

    /**
     * Sets The type of verification performed on the identity.
     * @param {String} verification The type of verification performed on the identity.
     */
  }, {
    key: "setVerification",
    value: function setVerification(verification) {
      this['verification'] = verification;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Identity</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Identity} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Identity} The populated <code>Identity</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Identity();
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
        }
        if (data.hasOwnProperty('verification')) {
          obj['verification'] = _ApiClient["default"].convertToType(data['verification'], 'String');
        }
      }
      return obj;
    }
  }]);
  return Identity;
}();
/**
 * The type of identity.
 * @member {String} type
 */
Identity.prototype['type'] = undefined;

/**
 * The identity value.
 * @member {String} value
 */
Identity.prototype['value'] = undefined;

/**
 * The type of verification performed on the identity.
 * @member {String} verification
 */
Identity.prototype['verification'] = undefined;
var _default = Identity;
exports["default"] = _default;
