// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/InlineObject.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The InlineObject model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/InlineObject
 * @version 9.14.0
 */
var InlineObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InlineObject</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/InlineObject
   * @param code {String} The authorization code received via /oauth/authorize
   * @param grantType {String} Must be set to the string `authorization_code`
   * @param clientId {String} Your integration’s unique identifier
   * @param clientSecret {String} Your integration’s secret
   */
  function InlineObject(code, grantType, clientId, clientSecret) {
    _classCallCheck(this, InlineObject);
    InlineObject.initialize(this, code, grantType, clientId, clientSecret);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(InlineObject, [{
    key: "getCode",
    value:
    /**
         * Returns The authorization code received via /oauth/authorize
         * @return {String}
         */
    function getCode() {
      return this.code;
    }

    /**
     * Sets The authorization code received via /oauth/authorize
     * @param {String} code The authorization code received via /oauth/authorize
     */
  }, {
    key: "setCode",
    value: function setCode(code) {
      this['code'] = code;
    }
    /**
         * Returns Must be set to the string `authorization_code`
         * @return {String}
         */
  }, {
    key: "getGrantType",
    value: function getGrantType() {
      return this.grant_type;
    }

    /**
     * Sets Must be set to the string `authorization_code`
     * @param {String} grantType Must be set to the string `authorization_code`
     */
  }, {
    key: "setGrantType",
    value: function setGrantType(grantType) {
      this['grant_type'] = grantType;
    }
    /**
         * Returns Your integration’s unique identifier
         * @return {String}
         */
  }, {
    key: "getClientId",
    value: function getClientId() {
      return this.client_id;
    }

    /**
     * Sets Your integration’s unique identifier
     * @param {String} clientId Your integration’s unique identifier
     */
  }, {
    key: "setClientId",
    value: function setClientId(clientId) {
      this['client_id'] = clientId;
    }
    /**
         * Returns Your integration’s secret
         * @return {String}
         */
  }, {
    key: "getClientSecret",
    value: function getClientSecret() {
      return this.client_secret;
    }

    /**
     * Sets Your integration’s secret
     * @param {String} clientSecret Your integration’s secret
     */
  }, {
    key: "setClientSecret",
    value: function setClientSecret(clientSecret) {
      this['client_secret'] = clientSecret;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, code, grantType, clientId, clientSecret) {
      obj['code'] = code;
      obj['grant_type'] = grantType;
      obj['client_id'] = clientId;
      obj['client_secret'] = clientSecret;
    }

    /**
     * Constructs a <code>InlineObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/InlineObject} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/InlineObject} The populated <code>InlineObject</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InlineObject();
        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'String');
        }
        if (data.hasOwnProperty('grant_type')) {
          obj['grant_type'] = _ApiClient["default"].convertToType(data['grant_type'], 'String');
        }
        if (data.hasOwnProperty('client_id')) {
          obj['client_id'] = _ApiClient["default"].convertToType(data['client_id'], 'String');
        }
        if (data.hasOwnProperty('client_secret')) {
          obj['client_secret'] = _ApiClient["default"].convertToType(data['client_secret'], 'String');
        }
      }
      return obj;
    }
  }]);
  return InlineObject;
}();
/**
 * The authorization code received via /oauth/authorize
 * @member {String} code
 */
InlineObject.prototype['code'] = undefined;

/**
 * Must be set to the string `authorization_code`
 * @member {String} grant_type
 */
InlineObject.prototype['grant_type'] = undefined;

/**
 * Your integration’s unique identifier
 * @member {String} client_id
 */
InlineObject.prototype['client_id'] = undefined;

/**
 * Your integration’s secret
 * @member {String} client_secret
 */
InlineObject.prototype['client_secret'] = undefined;
var _default = InlineObject;
exports["default"] = _default;
