// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Integration = _interopRequireDefault(require("./Integration"));
var _Links = _interopRequireDefault(require("./Links"));
var _Meta = _interopRequireDefault(require("./Meta"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The IntegrationListResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListResponse
 * @version 9.14.0
 */
var IntegrationListResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IntegrationListResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListResponse
   */
  function IntegrationListResponse() {
    _classCallCheck(this, IntegrationListResponse);
    IntegrationListResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(IntegrationListResponse, [{
    key: "getIntegrations",
    value:
    /**
         * Returns List of returned integrations.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Integration>}
         */
    function getIntegrations() {
      return this.integrations;
    }

    /**
     * Sets List of returned integrations.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Integration>} integrations List of returned integrations.
     */
  }, {
    key: "setIntegrations",
    value: function setIntegrations(integrations) {
      this['integrations'] = integrations;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta}
         */
  }, {
    key: "getMeta",
    value: function getMeta() {
      return this.meta;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta} meta
     */
  }, {
    key: "setMeta",
    value: function setMeta(meta) {
      this['meta'] = meta;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Links}
         */
  }, {
    key: "getLinks",
    value: function getLinks() {
      return this.links;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} links
     */
  }, {
    key: "setLinks",
    value: function setLinks(links) {
      this['links'] = links;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>IntegrationListResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListResponse} The populated <code>IntegrationListResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IntegrationListResponse();
        if (data.hasOwnProperty('integrations')) {
          obj['integrations'] = _ApiClient["default"].convertToType(data['integrations'], [_Integration["default"]]);
        }
        if (data.hasOwnProperty('meta')) {
          obj['meta'] = _Meta["default"].constructFromObject(data['meta']);
        }
        if (data.hasOwnProperty('links')) {
          obj['links'] = _Links["default"].constructFromObject(data['links']);
        }
      }
      return obj;
    }
  }]);
  return IntegrationListResponse;
}();
/**
 * List of returned integrations.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Integration>} integrations
 */
IntegrationListResponse.prototype['integrations'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta} meta
 */
IntegrationListResponse.prototype['meta'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} links
 */
IntegrationListResponse.prototype['links'] = undefined;
var _default = IntegrationListResponse;
exports["default"] = _default;
