// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Links.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Links model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Links
 * @version 9.14.0
 */
var Links = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Links</code>.
   * Previous and next page links, if they exist.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Links
   */
  function Links() {
    _classCallCheck(this, Links);
    Links.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Links, [{
    key: "getPrev",
    value:
    /**
         * Returns A link to the previous page. Not specified if there are no previous pages.
         * @return {String}
         */
    function getPrev() {
      return this.prev;
    }

    /**
     * Sets A link to the previous page. Not specified if there are no previous pages.
     * @param {String} prev A link to the previous page. Not specified if there are no previous pages.
     */
  }, {
    key: "setPrev",
    value: function setPrev(prev) {
      this['prev'] = prev;
    }
    /**
         * Returns A link to the next page. Not specified if there are no subsequent pages.
         * @return {String}
         */
  }, {
    key: "getNext",
    value: function getNext() {
      return this.next;
    }

    /**
     * Sets A link to the next page. Not specified if there are no subsequent pages.
     * @param {String} next A link to the next page. Not specified if there are no subsequent pages.
     */
  }, {
    key: "setNext",
    value: function setNext(next) {
      this['next'] = next;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Links</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} The populated <code>Links</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Links();
        if (data.hasOwnProperty('prev')) {
          obj['prev'] = _ApiClient["default"].convertToType(data['prev'], 'String');
        }
        if (data.hasOwnProperty('next')) {
          obj['next'] = _ApiClient["default"].convertToType(data['next'], 'String');
        }
      }
      return obj;
    }
  }]);
  return Links;
}();
/**
 * A link to the previous page. Not specified if there are no previous pages.
 * @member {String} prev
 */
Links.prototype['prev'] = undefined;

/**
 * A link to the next page. Not specified if there are no subsequent pages.
 * @member {String} next
 */
Links.prototype['next'] = undefined;
var _default = Links;
exports["default"] = _default;
