// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageCoordinates.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The LocationMessageCoordinates model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageCoordinates
 * @version 9.14.0
 */
var LocationMessageCoordinates = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LocationMessageCoordinates</code>.
   * The coordinates of the location.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageCoordinates
   * @param lat {Number} Global latitude.
   * @param _long {Number} Global longitude.
   */
  function LocationMessageCoordinates(lat, _long) {
    _classCallCheck(this, LocationMessageCoordinates);
    LocationMessageCoordinates.initialize(this, lat, _long);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(LocationMessageCoordinates, [{
    key: "getLat",
    value:
    /**
         * Returns Global latitude.
         * @return {Number}
         */
    function getLat() {
      return this.lat;
    }

    /**
     * Sets Global latitude.
     * @param {Number} lat Global latitude.
     */
  }, {
    key: "setLat",
    value: function setLat(lat) {
      this['lat'] = lat;
    }
    /**
         * Returns Global longitude.
         * @return {Number}
         */
  }, {
    key: "getLong",
    value: function getLong() {
      return this["long"];
    }

    /**
     * Sets Global longitude.
     * @param {Number} _long Global longitude.
     */
  }, {
    key: "setLong",
    value: function setLong(_long) {
      this['long'] = _long;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, lat, _long) {
      obj['lat'] = lat;
      obj['long'] = _long;
    }

    /**
     * Constructs a <code>LocationMessageCoordinates</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageCoordinates} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageCoordinates} The populated <code>LocationMessageCoordinates</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LocationMessageCoordinates();
        if (data.hasOwnProperty('lat')) {
          obj['lat'] = _ApiClient["default"].convertToType(data['lat'], 'Number');
        }
        if (data.hasOwnProperty('long')) {
          obj['long'] = _ApiClient["default"].convertToType(data['long'], 'Number');
        }
      }
      return obj;
    }
  }]);
  return LocationMessageCoordinates;
}();
/**
 * Global latitude.
 * @member {Number} lat
 */
LocationMessageCoordinates.prototype['lat'] = undefined;

/**
 * Global longitude.
 * @member {Number} long
 */
LocationMessageCoordinates.prototype['long'] = undefined;
var _default = LocationMessageCoordinates;
exports["default"] = _default;
