// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _MessageOverrideApple = _interopRequireDefault(require("./MessageOverrideApple"));
var _MessageOverrideLine = _interopRequireDefault(require("./MessageOverrideLine"));
var _MessageOverrideMessenger = _interopRequireDefault(require("./MessageOverrideMessenger"));
var _MessageOverridePayload = _interopRequireDefault(require("./MessageOverridePayload"));
var _MessageOverrideWhatsapp = _interopRequireDefault(require("./MessageOverrideWhatsapp"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The MessageOverride model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride
 * @version 9.14.0
 */
var MessageOverride = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MessageOverride</code>.
   * A raw payload containing a message that is sent directly to a channel. Messages for [Apple Business Chat](https://docs.smooch.io/guide/apple-business-chat/#passthrough-api), [Line](https://docs.smooch.io/guide/line/#passthrough-api), [Messenger](https://docs.smooch.io/guide/facebook-messenger/#passthrough-api), [Whatsapp](https://docs.smooch.io/guide/whatsapp/#passthrough-api) channels are supported.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideApple
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideLine
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideMessenger
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideWhatsapp
   */
  function MessageOverride() {
    _classCallCheck(this, MessageOverride);
    _MessageOverrideApple["default"].initialize(this);
    _MessageOverrideLine["default"].initialize(this);
    _MessageOverrideMessenger["default"].initialize(this);
    _MessageOverrideWhatsapp["default"].initialize(this);
    MessageOverride.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(MessageOverride, [{
    key: "getApple",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload}
         */
    function getApple() {
      return this.apple;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} apple
     */
  }, {
    key: "setApple",
    value: function setApple(apple) {
      this['apple'] = apple;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload}
         */
  }, {
    key: "getLine",
    value: function getLine() {
      return this.line;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} line
     */
  }, {
    key: "setLine",
    value: function setLine(line) {
      this['line'] = line;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload}
         */
  }, {
    key: "getMessenger",
    value: function getMessenger() {
      return this.messenger;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} messenger
     */
  }, {
    key: "setMessenger",
    value: function setMessenger(messenger) {
      this['messenger'] = messenger;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload}
         */
  }, {
    key: "getWhatsapp",
    value: function getWhatsapp() {
      return this.whatsapp;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} whatsapp
     */
  }, {
    key: "setWhatsapp",
    value: function setWhatsapp(whatsapp) {
      this['whatsapp'] = whatsapp;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>MessageOverride</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverride} The populated <code>MessageOverride</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MessageOverride();
        _MessageOverrideApple["default"].constructFromObject(data, obj);
        _MessageOverrideLine["default"].constructFromObject(data, obj);
        _MessageOverrideMessenger["default"].constructFromObject(data, obj);
        _MessageOverrideWhatsapp["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('apple')) {
          obj['apple'] = _MessageOverridePayload["default"].constructFromObject(data['apple']);
        }
        if (data.hasOwnProperty('line')) {
          obj['line'] = _MessageOverridePayload["default"].constructFromObject(data['line']);
        }
        if (data.hasOwnProperty('messenger')) {
          obj['messenger'] = _MessageOverridePayload["default"].constructFromObject(data['messenger']);
        }
        if (data.hasOwnProperty('whatsapp')) {
          obj['whatsapp'] = _MessageOverridePayload["default"].constructFromObject(data['whatsapp']);
        }
      }
      return obj;
    }
  }]);
  return MessageOverride;
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} apple
 */
MessageOverride.prototype['apple'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} line
 */
MessageOverride.prototype['line'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} messenger
 */
MessageOverride.prototype['messenger'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} whatsapp
 */
MessageOverride.prototype['whatsapp'] = undefined;

// Implement MessageOverrideApple interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} apple
 */
_MessageOverrideApple["default"].prototype['apple'] = undefined;
// Implement MessageOverrideLine interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} line
 */
_MessageOverrideLine["default"].prototype['line'] = undefined;
// Implement MessageOverrideMessenger interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} messenger
 */
_MessageOverrideMessenger["default"].prototype['messenger'] = undefined;
// Implement MessageOverrideWhatsapp interface:
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} whatsapp
 */
_MessageOverrideWhatsapp["default"].prototype['whatsapp'] = undefined;
var _default = MessageOverride;
exports["default"] = _default;
