// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/PrechatCapture.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Field = _interopRequireDefault(require("./Field"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The PrechatCapture model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/PrechatCapture
 * @version 9.14.0
 */
var PrechatCapture = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PrechatCapture</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/PrechatCapture
   */
  function PrechatCapture() {
    _classCallCheck(this, PrechatCapture);
    PrechatCapture.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(PrechatCapture, [{
    key: "getAvatarUrl",
    value:
    /**
         * Returns Sets the URL of the avatar to use for the automatic reply to the prechat capture messages.
         * @return {String}
         */
    function getAvatarUrl() {
      return this.avatarUrl;
    }

    /**
     * Sets Sets the URL of the avatar to use for the automatic reply to the prechat capture messages.
     * @param {String} avatarUrl Sets the URL of the avatar to use for the automatic reply to the prechat capture messages.
     */
  }, {
    key: "setAvatarUrl",
    value: function setAvatarUrl(avatarUrl) {
      this['avatarUrl'] = avatarUrl;
    }
    /**
         * Returns If true, enables the Prechat Capture add-on when the Web Messenger is initialized.
         * @return {Boolean}
         */
  }, {
    key: "getEnabled",
    value: function getEnabled() {
      return this.enabled;
    }

    /**
     * Sets If true, enables the Prechat Capture add-on when the Web Messenger is initialized.
     * @param {Boolean} enabled If true, enables the Prechat Capture add-on when the Web Messenger is initialized.
     */
  }, {
    key: "setEnabled",
    value: function setEnabled(enabled) {
      this['enabled'] = enabled;
    }
    /**
         * Returns f true and Mailgun is integrated, will automatically link submitted emails.
         * @return {Boolean}
         */
  }, {
    key: "getEnableEmailLinking",
    value: function getEnableEmailLinking() {
      return this.enableEmailLinking;
    }

    /**
     * Sets f true and Mailgun is integrated, will automatically link submitted emails.
     * @param {Boolean} enableEmailLinking f true and Mailgun is integrated, will automatically link submitted emails.
     */
  }, {
    key: "setEnableEmailLinking",
    value: function setEnableEmailLinking(enableEmailLinking) {
      this['enableEmailLinking'] = enableEmailLinking;
    }
    /**
         * Returns Array of Fields. Overrides the default Prechat Capture fields to define a custom form.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Field>}
         */
  }, {
    key: "getFields",
    value: function getFields() {
      return this.fields;
    }

    /**
     * Sets Array of Fields. Overrides the default Prechat Capture fields to define a custom form.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Field>} fields Array of Fields. Overrides the default Prechat Capture fields to define a custom form.
     */
  }, {
    key: "setFields",
    value: function setFields(fields) {
      this['fields'] = fields;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>PrechatCapture</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/PrechatCapture} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/PrechatCapture} The populated <code>PrechatCapture</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PrechatCapture();
        if (data.hasOwnProperty('avatarUrl')) {
          obj['avatarUrl'] = _ApiClient["default"].convertToType(data['avatarUrl'], 'String');
        }
        if (data.hasOwnProperty('enabled')) {
          obj['enabled'] = _ApiClient["default"].convertToType(data['enabled'], 'Boolean');
        }
        if (data.hasOwnProperty('enableEmailLinking')) {
          obj['enableEmailLinking'] = _ApiClient["default"].convertToType(data['enableEmailLinking'], 'Boolean');
        }
        if (data.hasOwnProperty('fields')) {
          obj['fields'] = _ApiClient["default"].convertToType(data['fields'], [_Field["default"]]);
        }
      }
      return obj;
    }
  }]);
  return PrechatCapture;
}();
/**
 * Sets the URL of the avatar to use for the automatic reply to the prechat capture messages.
 * @member {String} avatarUrl
 * @default 'undefined'
 */
PrechatCapture.prototype['avatarUrl'] = 'undefined';

/**
 * If true, enables the Prechat Capture add-on when the Web Messenger is initialized.
 * @member {Boolean} enabled
 * @default false
 */
PrechatCapture.prototype['enabled'] = false;

/**
 * f true and Mailgun is integrated, will automatically link submitted emails.
 * @member {Boolean} enableEmailLinking
 * @default false
 */
PrechatCapture.prototype['enableEmailLinking'] = false;

/**
 * Array of Fields. Overrides the default Prechat Capture fields to define a custom form.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Field>} fields
 */
PrechatCapture.prototype['fields'] = undefined;
var _default = PrechatCapture;
exports["default"] = _default;
