// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Message = _interopRequireDefault(require("./Message"));
var _QuotedMessageExternalMessageId = _interopRequireDefault(require("./QuotedMessageExternalMessageId"));
var _QuotedMessageMessage = _interopRequireDefault(require("./QuotedMessageMessage"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The QuotedMessage model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessage
 * @version 9.14.0
 */
var QuotedMessage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>QuotedMessage</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessage
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessageMessage
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessageExternalMessageId
   */
  function QuotedMessage() {
    _classCallCheck(this, QuotedMessage);
    _QuotedMessageMessage["default"].initialize(this);
    _QuotedMessageExternalMessageId["default"].initialize(this);
    QuotedMessage.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(QuotedMessage, [{
    key: "getType",
    value:
    /**
         * Returns The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
     * @param {String} type The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Message}
         */
  }, {
    key: "getMessage",
    value: function getMessage() {
      return this.message;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Message} message
     */
  }, {
    key: "setMessage",
    value: function setMessage(message) {
      this['message'] = message;
    }
    /**
         * Returns The external message Id of the quoted message.
         * @return {String}
         */
  }, {
    key: "getExternalMessageId",
    value: function getExternalMessageId() {
      return this.externalMessageId;
    }

    /**
     * Sets The external message Id of the quoted message.
     * @param {String} externalMessageId The external message Id of the quoted message.
     */
  }, {
    key: "setExternalMessageId",
    value: function setExternalMessageId(externalMessageId) {
      this['externalMessageId'] = externalMessageId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>QuotedMessage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessage} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/QuotedMessage} The populated <code>QuotedMessage</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new QuotedMessage();
        _QuotedMessageMessage["default"].constructFromObject(data, obj);
        _QuotedMessageExternalMessageId["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], _Message["default"]);
        }
        if (data.hasOwnProperty('externalMessageId')) {
          obj['externalMessageId'] = _ApiClient["default"].convertToType(data['externalMessageId'], 'String');
        }
      }
      return obj;
    }
  }]);
  return QuotedMessage;
}();
/**
 * The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
 * @member {String} type
 * @default 'externalMessageId'
 */
QuotedMessage.prototype['type'] = 'externalMessageId';

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Message} message
 */
QuotedMessage.prototype['message'] = undefined;

/**
 * The external message Id of the quoted message.
 * @member {String} externalMessageId
 */
QuotedMessage.prototype['externalMessageId'] = undefined;

// Implement QuotedMessageMessage interface:
/**
 * The type of quotedMessage - a complete Sunshine Conversations message is provided.
 * @member {String} type
 * @default 'message'
 */
_QuotedMessageMessage["default"].prototype['type'] = 'message';
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Message} message
 */
_QuotedMessageMessage["default"].prototype['message'] = undefined;
// Implement QuotedMessageExternalMessageId interface:
/**
 * The type of quotedMessage - `externalMessageId` if no Sunshine Conversations message matched the quoted message.
 * @member {String} type
 * @default 'externalMessageId'
 */
_QuotedMessageExternalMessageId["default"].prototype['type'] = 'externalMessageId';
/**
 * The external message Id of the quoted message.
 * @member {String} externalMessageId
 */
_QuotedMessageExternalMessageId["default"].prototype['externalMessageId'] = undefined;
var _default = QuotedMessage;
exports["default"] = _default;
