// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Telegram.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Integration = _interopRequireDefault(require("./Integration"));
var _Status = _interopRequireDefault(require("./Status"));
var _TelegramAllOf = _interopRequireDefault(require("./TelegramAllOf"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Telegram model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Telegram
 * @version 9.14.0
 */
var Telegram = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Telegram</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Telegram
   * @extends module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/TelegramAllOf
   * @param type {String} To configure a Telegram integration, acquire the required information from the user and call the Create Integration endpoint. 
   * @param token {String} Telegram Bot Token.
   */
  function Telegram(type, token) {
    _classCallCheck(this, Telegram);
    _Integration["default"].initialize(this, type);
    _TelegramAllOf["default"].initialize(this, token);
    Telegram.initialize(this, type, token);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Telegram, [{
    key: "getType",
    value:
    /**
         * Returns To configure a Telegram integration, acquire the required information from the user and call the Create Integration endpoint. 
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets To configure a Telegram integration, acquire the required information from the user and call the Create Integration endpoint. 
     * @param {String} type To configure a Telegram integration, acquire the required information from the user and call the Create Integration endpoint. 
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns Telegram Bot Token.
         * @return {String}
         */
  }, {
    key: "getToken",
    value: function getToken() {
      return this.token;
    }

    /**
     * Sets Telegram Bot Token.
     * @param {String} token Telegram Bot Token.
     */
  }, {
    key: "setToken",
    value: function setToken(token) {
      this['token'] = token;
    }
    /**
         * Returns Username of the botId
         * @return {String}
         */
  }, {
    key: "getUsername",
    value: function getUsername() {
      return this.username;
    }

    /**
     * Sets Username of the botId
     * @param {String} username Username of the botId
     */
  }, {
    key: "setUsername",
    value: function setUsername(username) {
      this['username'] = username;
    }
    /**
         * Returns A human-friendly name used to identify the integration.
         * @return {String}
         */
  }, {
    key: "getBotId",
    value: function getBotId() {
      return this.botId;
    }

    /**
     * Sets A human-friendly name used to identify the integration.
     * @param {String} botId A human-friendly name used to identify the integration.
     */
  }, {
    key: "setBotId",
    value: function setBotId(botId) {
      this['botId'] = botId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, token) {
      obj['token'] = token;
    }

    /**
     * Constructs a <code>Telegram</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Telegram} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Telegram} The populated <code>Telegram</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Telegram();
        _Integration["default"].constructFromObject(data, obj);
        _Integration["default"].constructFromObject(data, obj);
        _TelegramAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }
        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
        if (data.hasOwnProperty('botId')) {
          obj['botId'] = _ApiClient["default"].convertToType(data['botId'], 'String');
        }
      }
      return obj;
    }
  }]);
  return Telegram;
}();
/**
 * To configure a Telegram integration, acquire the required information from the user and call the Create Integration endpoint. 
 * @member {String} type
 * @default 'telegram'
 */
Telegram.prototype['type'] = 'telegram';

/**
 * Telegram Bot Token.
 * @member {String} token
 */
Telegram.prototype['token'] = undefined;

/**
 * Username of the botId
 * @member {String} username
 */
Telegram.prototype['username'] = undefined;

/**
 * A human-friendly name used to identify the integration.
 * @member {String} botId
 */
Telegram.prototype['botId'] = undefined;

// Implement Integration interface:
/**
 * The unique ID of the integration.
 * @member {String} id
 */
_Integration["default"].prototype['id'] = undefined;
/**
 * @member {String} type
 */
_Integration["default"].prototype['type'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Status} status
 */
_Integration["default"].prototype['status'] = undefined;
/**
 * A human-friendly name used to identify the integration.
 * @member {String} displayName
 */
_Integration["default"].prototype['displayName'] = undefined;
// Implement TelegramAllOf interface:
/**
 * To configure a Telegram integration, acquire the required information from the user and call the Create Integration endpoint. 
 * @member {String} type
 * @default 'telegram'
 */
_TelegramAllOf["default"].prototype['type'] = 'telegram';
/**
 * Telegram Bot Token.
 * @member {String} token
 */
_TelegramAllOf["default"].prototype['token'] = undefined;
/**
 * Username of the botId
 * @member {String} username
 */
_TelegramAllOf["default"].prototype['username'] = undefined;
/**
 * A human-friendly name used to identify the integration.
 * @member {String} botId
 */
_TelegramAllOf["default"].prototype['botId'] = undefined;
var _default = Telegram;
exports["default"] = _default;
