// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Twilio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Integration = _interopRequireDefault(require("./Integration"));
var _Status = _interopRequireDefault(require("./Status"));
var _TwilioAllOf = _interopRequireDefault(require("./TwilioAllOf"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Twilio model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Twilio
 * @version 9.14.0
 */
var Twilio = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Twilio</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Twilio
   * @extends module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/Integration
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/TwilioAllOf
   * @param type {String} To configure a Twilio integration, acquire the required information from the user and call the Create Integration endpoint. 
   * @param accountSid {String} Twilio Account SID.
   * @param authToken {String} Twilio Auth Token.
   */
  function Twilio(type, accountSid, authToken) {
    _classCallCheck(this, Twilio);
    _Integration["default"].initialize(this, type);
    _TwilioAllOf["default"].initialize(this, accountSid, authToken);
    Twilio.initialize(this, type, accountSid, authToken);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Twilio, [{
    key: "getType",
    value:
    /**
         * Returns To configure a Twilio integration, acquire the required information from the user and call the Create Integration endpoint. 
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets To configure a Twilio integration, acquire the required information from the user and call the Create Integration endpoint. 
     * @param {String} type To configure a Twilio integration, acquire the required information from the user and call the Create Integration endpoint. 
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns Twilio Account SID.
         * @return {String}
         */
  }, {
    key: "getAccountSid",
    value: function getAccountSid() {
      return this.accountSid;
    }

    /**
     * Sets Twilio Account SID.
     * @param {String} accountSid Twilio Account SID.
     */
  }, {
    key: "setAccountSid",
    value: function setAccountSid(accountSid) {
      this['accountSid'] = accountSid;
    }
    /**
         * Returns Twilio Auth Token.
         * @return {String}
         */
  }, {
    key: "getAuthToken",
    value: function getAuthToken() {
      return this.authToken;
    }

    /**
     * Sets Twilio Auth Token.
     * @param {String} authToken Twilio Auth Token.
     */
  }, {
    key: "setAuthToken",
    value: function setAuthToken(authToken) {
      this['authToken'] = authToken;
    }
    /**
         * Returns SID for specific phone number. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
         * @return {String}
         */
  }, {
    key: "getPhoneNumberSid",
    value: function getPhoneNumberSid() {
      return this.phoneNumberSid;
    }

    /**
     * Sets SID for specific phone number. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
     * @param {String} phoneNumberSid SID for specific phone number. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
     */
  }, {
    key: "setPhoneNumberSid",
    value: function setPhoneNumberSid(phoneNumberSid) {
      this['phoneNumberSid'] = phoneNumberSid;
    }
    /**
         * Returns SID for specific messaging service. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
         * @return {String}
         */
  }, {
    key: "getMessagingServiceSid",
    value: function getMessagingServiceSid() {
      return this.messagingServiceSid;
    }

    /**
     * Sets SID for specific messaging service. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
     * @param {String} messagingServiceSid SID for specific messaging service. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
     */
  }, {
    key: "setMessagingServiceSid",
    value: function setMessagingServiceSid(messagingServiceSid) {
      this['messagingServiceSid'] = messagingServiceSid;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, accountSid, authToken) {
      obj['accountSid'] = accountSid;
      obj['authToken'] = authToken;
    }

    /**
     * Constructs a <code>Twilio</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Twilio} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Twilio} The populated <code>Twilio</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Twilio();
        _Integration["default"].constructFromObject(data, obj);
        _Integration["default"].constructFromObject(data, obj);
        _TwilioAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('accountSid')) {
          obj['accountSid'] = _ApiClient["default"].convertToType(data['accountSid'], 'String');
        }
        if (data.hasOwnProperty('authToken')) {
          obj['authToken'] = _ApiClient["default"].convertToType(data['authToken'], 'String');
        }
        if (data.hasOwnProperty('phoneNumberSid')) {
          obj['phoneNumberSid'] = _ApiClient["default"].convertToType(data['phoneNumberSid'], 'String');
        }
        if (data.hasOwnProperty('messagingServiceSid')) {
          obj['messagingServiceSid'] = _ApiClient["default"].convertToType(data['messagingServiceSid'], 'String');
        }
      }
      return obj;
    }
  }]);
  return Twilio;
}();
/**
 * To configure a Twilio integration, acquire the required information from the user and call the Create Integration endpoint. 
 * @member {String} type
 * @default 'twilio'
 */
Twilio.prototype['type'] = 'twilio';

/**
 * Twilio Account SID.
 * @member {String} accountSid
 */
Twilio.prototype['accountSid'] = undefined;

/**
 * Twilio Auth Token.
 * @member {String} authToken
 */
Twilio.prototype['authToken'] = undefined;

/**
 * SID for specific phone number. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
 * @member {String} phoneNumberSid
 */
Twilio.prototype['phoneNumberSid'] = undefined;

/**
 * SID for specific messaging service. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
 * @member {String} messagingServiceSid
 */
Twilio.prototype['messagingServiceSid'] = undefined;

// Implement Integration interface:
/**
 * The unique ID of the integration.
 * @member {String} id
 */
_Integration["default"].prototype['id'] = undefined;
/**
 * @member {String} type
 */
_Integration["default"].prototype['type'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Status} status
 */
_Integration["default"].prototype['status'] = undefined;
/**
 * A human-friendly name used to identify the integration.
 * @member {String} displayName
 */
_Integration["default"].prototype['displayName'] = undefined;
// Implement TwilioAllOf interface:
/**
 * To configure a Twilio integration, acquire the required information from the user and call the Create Integration endpoint. 
 * @member {String} type
 * @default 'twilio'
 */
_TwilioAllOf["default"].prototype['type'] = 'twilio';
/**
 * Twilio Account SID.
 * @member {String} accountSid
 */
_TwilioAllOf["default"].prototype['accountSid'] = undefined;
/**
 * Twilio Auth Token.
 * @member {String} authToken
 */
_TwilioAllOf["default"].prototype['authToken'] = undefined;
/**
 * SID for specific phone number. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
 * @member {String} phoneNumberSid
 */
_TwilioAllOf["default"].prototype['phoneNumberSid'] = undefined;
/**
 * SID for specific messaging service. One of `messagingServiceSid` or `phoneNumberSid` must be provided when creating a Twilio integration.
 * @member {String} messagingServiceSid
 */
_TwilioAllOf["default"].prototype['messagingServiceSid'] = undefined;
var _default = Twilio;
exports["default"] = _default;
