// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedConversations.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ConversationTruncated = _interopRequireDefault(require("./ConversationTruncated"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The UserMergeEventAllOfPayloadMergedConversations model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedConversations
 * @version 9.14.0
 */
var UserMergeEventAllOfPayloadMergedConversations = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserMergeEventAllOfPayloadMergedConversations</code>.
   * Contains information about the conversations that were merged together as a result of the operation, if applicable. If no conversations were merged, this property is omitted.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedConversations
   */
  function UserMergeEventAllOfPayloadMergedConversations() {
    _classCallCheck(this, UserMergeEventAllOfPayloadMergedConversations);
    UserMergeEventAllOfPayloadMergedConversations.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(UserMergeEventAllOfPayloadMergedConversations, [{
    key: "getSurviving",
    value:
    /**
         * Returns The conversation that now represents the merged conversation object.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated}
         */
    function getSurviving() {
      return this.surviving;
    }

    /**
     * Sets The conversation that now represents the merged conversation object.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} surviving The conversation that now represents the merged conversation object.
     */
  }, {
    key: "setSurviving",
    value: function setSurviving(surviving) {
      this['surviving'] = surviving;
    }
    /**
         * Returns The conversation that was unified into the surviving conversation object.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated}
         */
  }, {
    key: "getDiscarded",
    value: function getDiscarded() {
      return this.discarded;
    }

    /**
     * Sets The conversation that was unified into the surviving conversation object.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} discarded The conversation that was unified into the surviving conversation object.
     */
  }, {
    key: "setDiscarded",
    value: function setDiscarded(discarded) {
      this['discarded'] = discarded;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UserMergeEventAllOfPayloadMergedConversations</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedConversations} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedConversations} The populated <code>UserMergeEventAllOfPayloadMergedConversations</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserMergeEventAllOfPayloadMergedConversations();
        if (data.hasOwnProperty('surviving')) {
          obj['surviving'] = _ApiClient["default"].convertToType(data['surviving'], _ConversationTruncated["default"]);
        }
        if (data.hasOwnProperty('discarded')) {
          obj['discarded'] = _ApiClient["default"].convertToType(data['discarded'], _ConversationTruncated["default"]);
        }
      }
      return obj;
    }
  }]);
  return UserMergeEventAllOfPayloadMergedConversations;
}();
/**
 * The conversation that now represents the merged conversation object.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} surviving
 */
UserMergeEventAllOfPayloadMergedConversations.prototype['surviving'] = undefined;

/**
 * The conversation that was unified into the surviving conversation object.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} discarded
 */
UserMergeEventAllOfPayloadMergedConversations.prototype['discarded'] = undefined;
var _default = UserMergeEventAllOfPayloadMergedConversations;
exports["default"] = _default;
