// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/WebhookBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The WebhookBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/WebhookBody
 * @version 9.14.0
 */
var WebhookBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebhookBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/WebhookBody
   */
  function WebhookBody() {
    _classCallCheck(this, WebhookBody);
    WebhookBody.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(WebhookBody, [{
    key: "getTarget",
    value:
    /**
         * Returns URL to be called when the webhook is triggered.
         * @return {String}
         */
    function getTarget() {
      return this.target;
    }

    /**
     * Sets URL to be called when the webhook is triggered.
     * @param {String} target URL to be called when the webhook is triggered.
     */
  }, {
    key: "setTarget",
    value: function setTarget(target) {
      this['target'] = target;
    }
    /**
         * Returns An array of triggers the integration is subscribed to. This property is case sensitive. [More details](https://docs.smooch.io/rest/#section/Webhook-Triggers).
         * @return {Array.<String>}
         */
  }, {
    key: "getTriggers",
    value: function getTriggers() {
      return this.triggers;
    }

    /**
     * Sets An array of triggers the integration is subscribed to. This property is case sensitive. [More details](https://docs.smooch.io/rest/#section/Webhook-Triggers).
     * @param {Array.<String>} triggers An array of triggers the integration is subscribed to. This property is case sensitive. [More details](https://docs.smooch.io/rest/#section/Webhook-Triggers).
     */
  }, {
    key: "setTriggers",
    value: function setTriggers(triggers) {
      this['triggers'] = triggers;
    }
    /**
         * Returns A boolean specifying whether webhook payloads should include the complete user schema for events involving a specific user.
         * @return {Boolean}
         */
  }, {
    key: "getIncludeFullUser",
    value: function getIncludeFullUser() {
      return this.includeFullUser;
    }

    /**
     * Sets A boolean specifying whether webhook payloads should include the complete user schema for events involving a specific user.
     * @param {Boolean} includeFullUser A boolean specifying whether webhook payloads should include the complete user schema for events involving a specific user.
     */
  }, {
    key: "setIncludeFullUser",
    value: function setIncludeFullUser(includeFullUser) {
      this['includeFullUser'] = includeFullUser;
    }
    /**
         * Returns A boolean specifying whether webhook payloads should include the client and device object (when applicable).
         * @return {Boolean}
         */
  }, {
    key: "getIncludeFullSource",
    value: function getIncludeFullSource() {
      return this.includeFullSource;
    }

    /**
     * Sets A boolean specifying whether webhook payloads should include the client and device object (when applicable).
     * @param {Boolean} includeFullSource A boolean specifying whether webhook payloads should include the client and device object (when applicable).
     */
  }, {
    key: "setIncludeFullSource",
    value: function setIncludeFullSource(includeFullSource) {
      this['includeFullSource'] = includeFullSource;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>WebhookBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/WebhookBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/WebhookBody} The populated <code>WebhookBody</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhookBody();
        if (data.hasOwnProperty('target')) {
          obj['target'] = _ApiClient["default"].convertToType(data['target'], 'String');
        }
        if (data.hasOwnProperty('triggers')) {
          obj['triggers'] = _ApiClient["default"].convertToType(data['triggers'], ['String']);
        }
        if (data.hasOwnProperty('includeFullUser')) {
          obj['includeFullUser'] = _ApiClient["default"].convertToType(data['includeFullUser'], 'Boolean');
        }
        if (data.hasOwnProperty('includeFullSource')) {
          obj['includeFullSource'] = _ApiClient["default"].convertToType(data['includeFullSource'], 'Boolean');
        }
      }
      return obj;
    }
  }]);
  return WebhookBody;
}();
/**
 * URL to be called when the webhook is triggered.
 * @member {String} target
 */
WebhookBody.prototype['target'] = undefined;

/**
 * An array of triggers the integration is subscribed to. This property is case sensitive. [More details](https://docs.smooch.io/rest/#section/Webhook-Triggers).
 * @member {Array.<String>} triggers
 */
WebhookBody.prototype['triggers'] = undefined;

/**
 * A boolean specifying whether webhook payloads should include the complete user schema for events involving a specific user.
 * @member {Boolean} includeFullUser
 * @default false
 */
WebhookBody.prototype['includeFullUser'] = false;

/**
 * A boolean specifying whether webhook payloads should include the client and device object (when applicable).
 * @member {Boolean} includeFullSource
 * @default false
 */
WebhookBody.prototype['includeFullSource'] = false;
var _default = WebhookBody;
exports["default"] = _default;
