// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Webview.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ExtraChannelOptions = _interopRequireDefault(require("./ExtraChannelOptions"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Webview model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Webview
 * @version 9.14.0
 */
var Webview = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Webview</code>.
   * When a webview actions is clicked/tapped, the provided URI will be loaded in a webview. Channels that do not support webviews will open the fallback URI instead.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Webview
   * @param type {String} The type of action.
   * @param uri {String} The webview URI. This is the URI that will open in the webview when clicking the button.
   * @param text {String} The button text.
   * @param fallback {String} The fallback uri for channels that don’t support webviews. Used for actions of type webview.
   */
  function Webview(type, uri, text, fallback) {
    _classCallCheck(this, Webview);
    Webview.initialize(this, type, uri, text, fallback);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Webview, [{
    key: "getType",
    value:
    /**
         * Returns The type of action.
         * @return {String}
         */
    function getType() {
      return this.type;
    }

    /**
     * Sets The type of action.
     * @param {String} type The type of action.
     */
  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The webview URI. This is the URI that will open in the webview when clicking the button.
         * @return {String}
         */
  }, {
    key: "getUri",
    value: function getUri() {
      return this.uri;
    }

    /**
     * Sets The webview URI. This is the URI that will open in the webview when clicking the button.
     * @param {String} uri The webview URI. This is the URI that will open in the webview when clicking the button.
     */
  }, {
    key: "setUri",
    value: function setUri(uri) {
      this['uri'] = uri;
    }
    /**
         * Returns The button text.
         * @return {String}
         */
  }, {
    key: "getText",
    value: function getText() {
      return this.text;
    }

    /**
     * Sets The button text.
     * @param {String} text The button text.
     */
  }, {
    key: "setText",
    value: function setText(text) {
      this['text'] = text;
    }
    /**
         * Returns Boolean value indicating whether the action is the default action for a message item in Facebook Messenger.
         * @return {Boolean}
         */
  }, {
    key: "getDefault",
    value: function getDefault() {
      return this["default"];
    }

    /**
     * Sets Boolean value indicating whether the action is the default action for a message item in Facebook Messenger.
     * @param {Boolean} _default Boolean value indicating whether the action is the default action for a message item in Facebook Messenger.
     */
  }, {
    key: "setDefault",
    value: function setDefault(_default) {
      this['default'] = _default;
    }
    /**
         * Returns Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
         * @return {Object}
         */
  }, {
    key: "getMetadata",
    value: function getMetadata() {
      return this.metadata;
    }

    /**
     * Sets Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
     * @param {Object} metadata Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
     */
  }, {
    key: "setMetadata",
    value: function setMetadata(metadata) {
      this['metadata'] = metadata;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ExtraChannelOptions}
         */
  }, {
    key: "getExtraChannelOptions",
    value: function getExtraChannelOptions() {
      return this.extraChannelOptions;
    }

    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ExtraChannelOptions} extraChannelOptions
     */
  }, {
    key: "setExtraChannelOptions",
    value: function setExtraChannelOptions(extraChannelOptions) {
      this['extraChannelOptions'] = extraChannelOptions;
    }
    /**
         * Returns The size to display a webview. Used for actions of type webview.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Webview.SizeEnum}
         */
  }, {
    key: "getSize",
    value: function getSize() {
      return this.size;
    }

    /**
     * Sets The size to display a webview. Used for actions of type webview.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Webview.SizeEnum} size The size to display a webview. Used for actions of type webview.
     */
  }, {
    key: "setSize",
    value: function setSize(size) {
      this['size'] = size;
    }
    /**
         * Returns The fallback uri for channels that don’t support webviews. Used for actions of type webview.
         * @return {String}
         */
  }, {
    key: "getFallback",
    value: function getFallback() {
      return this.fallback;
    }

    /**
     * Sets The fallback uri for channels that don’t support webviews. Used for actions of type webview.
     * @param {String} fallback The fallback uri for channels that don’t support webviews. Used for actions of type webview.
     */
  }, {
    key: "setFallback",
    value: function setFallback(fallback) {
      this['fallback'] = fallback;
    }
    /**
         * Returns Boolean value indicating if the webview should open automatically. Only one action per message can be set to true. Currently only supported on the Web Messenger.
         * @return {Boolean}
         */
  }, {
    key: "getOpenOnReceive",
    value: function getOpenOnReceive() {
      return this.openOnReceive;
    }

    /**
     * Sets Boolean value indicating if the webview should open automatically. Only one action per message can be set to true. Currently only supported on the Web Messenger.
     * @param {Boolean} openOnReceive Boolean value indicating if the webview should open automatically. Only one action per message can be set to true. Currently only supported on the Web Messenger.
     */
  }, {
    key: "setOpenOnReceive",
    value: function setOpenOnReceive(openOnReceive) {
      this['openOnReceive'] = openOnReceive;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, type, uri, text, fallback) {
      obj['type'] = type;
      obj['uri'] = uri;
      obj['text'] = text;
      obj['fallback'] = fallback;
    }

    /**
     * Constructs a <code>Webview</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Webview} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Webview} The populated <code>Webview</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Webview();
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('uri')) {
          obj['uri'] = _ApiClient["default"].convertToType(data['uri'], 'String');
        }
        if (data.hasOwnProperty('text')) {
          obj['text'] = _ApiClient["default"].convertToType(data['text'], 'String');
        }
        if (data.hasOwnProperty('default')) {
          obj['default'] = _ApiClient["default"].convertToType(data['default'], 'Boolean');
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
        if (data.hasOwnProperty('extraChannelOptions')) {
          obj['extraChannelOptions'] = _ExtraChannelOptions["default"].constructFromObject(data['extraChannelOptions']);
        }
        if (data.hasOwnProperty('size')) {
          obj['size'] = _ApiClient["default"].convertToType(data['size'], 'String');
        }
        if (data.hasOwnProperty('fallback')) {
          obj['fallback'] = _ApiClient["default"].convertToType(data['fallback'], 'String');
        }
        if (data.hasOwnProperty('openOnReceive')) {
          obj['openOnReceive'] = _ApiClient["default"].convertToType(data['openOnReceive'], 'Boolean');
        }
      }
      return obj;
    }
  }]);
  return Webview;
}();
/**
 * The type of action.
 * @member {String} type
 */
Webview.prototype['type'] = undefined;

/**
 * The webview URI. This is the URI that will open in the webview when clicking the button.
 * @member {String} uri
 */
Webview.prototype['uri'] = undefined;

/**
 * The button text.
 * @member {String} text
 */
Webview.prototype['text'] = undefined;

/**
 * Boolean value indicating whether the action is the default action for a message item in Facebook Messenger.
 * @member {Boolean} default
 */
Webview.prototype['default'] = undefined;

/**
 * Flat object containing custom properties. Strings, numbers and booleans  are the only supported format that can be passed to metadata. The metadata is limited to 4KB in size. 
 * @member {Object} metadata
 */
Webview.prototype['metadata'] = undefined;

/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ExtraChannelOptions} extraChannelOptions
 */
Webview.prototype['extraChannelOptions'] = undefined;

/**
 * The size to display a webview. Used for actions of type webview.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Webview.SizeEnum} size
 */
Webview.prototype['size'] = undefined;

/**
 * The fallback uri for channels that don’t support webviews. Used for actions of type webview.
 * @member {String} fallback
 */
Webview.prototype['fallback'] = undefined;

/**
 * Boolean value indicating if the webview should open automatically. Only one action per message can be set to true. Currently only supported on the Web Messenger.
 * @member {Boolean} openOnReceive
 */
Webview.prototype['openOnReceive'] = undefined;

/**
 * Allowed values for the <code>size</code> property.
 * @enum {String}
 * @readonly
 */
Webview['SizeEnum'] = {
  /**
   * value: "compact"
   * @const
   */
  "compact": "compact",
  /**
   * value: "tall"
   * @const
   */
  "tall": "tall",
  /**
   * value: "full"
   * @const
   */
  "full": "full"
};
var _default2 = Webview;
exports["default"] = _default2;
