// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.api/AppKeysApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AppKeyCreateBody = _interopRequireDefault(require("../sunshine-conversations-client.model/AppKeyCreateBody"));
var _AppKeyListResponse = _interopRequireDefault(require("../sunshine-conversations-client.model/AppKeyListResponse"));
var _AppKeyResponse = _interopRequireDefault(require("../sunshine-conversations-client.model/AppKeyResponse"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
* AppKeys service.
* @module sunshine-conversations-client/sunshine-conversations-client.api/AppKeysApi
* @version 9.14.0
*/
var AppKeysApi = /*#__PURE__*/function () {
  /**
  * Constructs a new AppKeysApi. 
  * @alias module:sunshine-conversations-client/sunshine-conversations-client.api/AppKeysApi
  * @class
  * @param {module:sunshine-conversations-client/ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:sunshine-conversations-client/ApiClient#instance} if unspecified.
  */
  function AppKeysApi(apiClient) {
    _classCallCheck(this, AppKeysApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Create App Key
   * Creates an API key for the specified app. The response body will include a secret  as well as its corresponding id, which you can use to generate JSON Web Tokens to  securely make API calls on behalf of the app. 
   * @param {String} appId Identifies the app.
   * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyCreateBody} appKeyCreateBody 
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyResponse} and HTTP response
   */
  _createClass(AppKeysApi, [{
    key: "createAppKeyWithHttpInfo",
    value: function createAppKeyWithHttpInfo(appId, appKeyCreateBody) {
      var postBody = appKeyCreateBody;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling createAppKey");
      }
      // verify the required parameter 'appKeyCreateBody' is set
      if (appKeyCreateBody === undefined || appKeyCreateBody === null) {
        throw new Error("Missing the required parameter 'appKeyCreateBody' when calling createAppKey");
      }
      var pathParams = {
        'appId': appId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _AppKeyResponse["default"];
      return this.apiClient.callApi('/v2/apps/{appId}/keys', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Create App Key
     * Creates an API key for the specified app. The response body will include a secret  as well as its corresponding id, which you can use to generate JSON Web Tokens to  securely make API calls on behalf of the app. 
     * @param {String} appId Identifies the app.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyCreateBody} appKeyCreateBody 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyResponse}
     */
  }, {
    key: "createAppKey",
    value: function createAppKey(appId, appKeyCreateBody) {
      return this.createAppKeyWithHttpInfo(appId, appKeyCreateBody).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Delete App Key
     * Removes an API key.
     * @param {String} appId Identifies the app.
     * @param {String} keyId The id of the key.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Object} and HTTP response
     */
  }, {
    key: "deleteAppKeyWithHttpInfo",
    value: function deleteAppKeyWithHttpInfo(appId, keyId) {
      var postBody = null;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling deleteAppKey");
      }
      // verify the required parameter 'keyId' is set
      if (keyId === undefined || keyId === null) {
        throw new Error("Missing the required parameter 'keyId' when calling deleteAppKey");
      }
      var pathParams = {
        'appId': appId,
        'keyId': keyId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = Object;
      return this.apiClient.callApi('/v2/apps/{appId}/keys/{keyId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Delete App Key
     * Removes an API key.
     * @param {String} appId Identifies the app.
     * @param {String} keyId The id of the key.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Object}
     */
  }, {
    key: "deleteAppKey",
    value: function deleteAppKey(appId, keyId) {
      return this.deleteAppKeyWithHttpInfo(appId, keyId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get App Key
     * Returns an API key.
     * @param {String} appId Identifies the app.
     * @param {String} keyId The id of the key.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyResponse} and HTTP response
     */
  }, {
    key: "getAppKeyWithHttpInfo",
    value: function getAppKeyWithHttpInfo(appId, keyId) {
      var postBody = null;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling getAppKey");
      }
      // verify the required parameter 'keyId' is set
      if (keyId === undefined || keyId === null) {
        throw new Error("Missing the required parameter 'keyId' when calling getAppKey");
      }
      var pathParams = {
        'appId': appId,
        'keyId': keyId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AppKeyResponse["default"];
      return this.apiClient.callApi('/v2/apps/{appId}/keys/{keyId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get App Key
     * Returns an API key.
     * @param {String} appId Identifies the app.
     * @param {String} keyId The id of the key.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyResponse}
     */
  }, {
    key: "getAppKey",
    value: function getAppKey(appId, keyId) {
      return this.getAppKeyWithHttpInfo(appId, keyId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * List App Keys
     * Lists all API keys for a given app.
     * @param {String} appId Identifies the app.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyListResponse} and HTTP response
     */
  }, {
    key: "listAppKeysWithHttpInfo",
    value: function listAppKeysWithHttpInfo(appId) {
      var postBody = null;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling listAppKeys");
      }
      var pathParams = {
        'appId': appId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AppKeyListResponse["default"];
      return this.apiClient.callApi('/v2/apps/{appId}/keys', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * List App Keys
     * Lists all API keys for a given app.
     * @param {String} appId Identifies the app.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/AppKeyListResponse}
     */
  }, {
    key: "listAppKeys",
    value: function listAppKeys(appId) {
      return this.listAppKeysWithHttpInfo(appId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
  return AppKeysApi;
}();
exports["default"] = AppKeysApi;
