// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.api/SwitchboardsApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _SwitchboardListResponse = _interopRequireDefault(require("../sunshine-conversations-client.model/SwitchboardListResponse"));
var _SwitchboardResponse = _interopRequireDefault(require("../sunshine-conversations-client.model/SwitchboardResponse"));
var _SwitchboardUpdateBody = _interopRequireDefault(require("../sunshine-conversations-client.model/SwitchboardUpdateBody"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Sunshine Conversations API
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 9.12.0
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * OpenAPI Generator version: 4.3.1
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
* Switchboards service.
* @module sunshine-conversations-client/sunshine-conversations-client.api/SwitchboardsApi
* @version 9.14.0
*/
var SwitchboardsApi = /*#__PURE__*/function () {
  /**
  * Constructs a new SwitchboardsApi. 
  * @alias module:sunshine-conversations-client/sunshine-conversations-client.api/SwitchboardsApi
  * @class
  * @param {module:sunshine-conversations-client/ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:sunshine-conversations-client/ApiClient#instance} if unspecified.
  */
  function SwitchboardsApi(apiClient) {
    _classCallCheck(this, SwitchboardsApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Create Switchboard
   * Create a switchboard.
   * @param {String} appId Identifies the app.
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardResponse} and HTTP response
   */
  _createClass(SwitchboardsApi, [{
    key: "createSwitchboardWithHttpInfo",
    value: function createSwitchboardWithHttpInfo(appId) {
      var postBody = null;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling createSwitchboard");
      }
      var pathParams = {
        'appId': appId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _SwitchboardResponse["default"];
      return this.apiClient.callApi('/v2/apps/{appId}/switchboards', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Create Switchboard
     * Create a switchboard.
     * @param {String} appId Identifies the app.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardResponse}
     */
  }, {
    key: "createSwitchboard",
    value: function createSwitchboard(appId) {
      return this.createSwitchboardWithHttpInfo(appId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Delete Switchboard
     * Deletes the switchboard and all its switchboard integrations. The integrations linked to these switchboard integrations are not deleted and will start receiving all conversation events.
     * @param {String} appId Identifies the app.
     * @param {String} switchboardId Identifies the switchboard.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Object} and HTTP response
     */
  }, {
    key: "deleteSwitchboardWithHttpInfo",
    value: function deleteSwitchboardWithHttpInfo(appId, switchboardId) {
      var postBody = null;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling deleteSwitchboard");
      }
      // verify the required parameter 'switchboardId' is set
      if (switchboardId === undefined || switchboardId === null) {
        throw new Error("Missing the required parameter 'switchboardId' when calling deleteSwitchboard");
      }
      var pathParams = {
        'appId': appId,
        'switchboardId': switchboardId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = Object;
      return this.apiClient.callApi('/v2/apps/{appId}/switchboards/{switchboardId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Delete Switchboard
     * Deletes the switchboard and all its switchboard integrations. The integrations linked to these switchboard integrations are not deleted and will start receiving all conversation events.
     * @param {String} appId Identifies the app.
     * @param {String} switchboardId Identifies the switchboard.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Object}
     */
  }, {
    key: "deleteSwitchboard",
    value: function deleteSwitchboard(appId, switchboardId) {
      return this.deleteSwitchboardWithHttpInfo(appId, switchboardId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * List Switchboards
     * Lists all switchboards belonging to the app. 
     * @param {String} appId Identifies the app.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardListResponse} and HTTP response
     */
  }, {
    key: "listSwitchboardsWithHttpInfo",
    value: function listSwitchboardsWithHttpInfo(appId) {
      var postBody = null;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling listSwitchboards");
      }
      var pathParams = {
        'appId': appId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _SwitchboardListResponse["default"];
      return this.apiClient.callApi('/v2/apps/{appId}/switchboards', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * List Switchboards
     * Lists all switchboards belonging to the app. 
     * @param {String} appId Identifies the app.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardListResponse}
     */
  }, {
    key: "listSwitchboards",
    value: function listSwitchboards(appId) {
      return this.listSwitchboardsWithHttpInfo(appId).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Update Switchboard
     * Updates a switchboard record.
     * @param {String} appId Identifies the app.
     * @param {String} switchboardId Identifies the switchboard.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardUpdateBody} switchboardUpdateBody 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardResponse} and HTTP response
     */
  }, {
    key: "updateSwitchboardWithHttpInfo",
    value: function updateSwitchboardWithHttpInfo(appId, switchboardId, switchboardUpdateBody) {
      var postBody = switchboardUpdateBody;
      // verify the required parameter 'appId' is set
      if (appId === undefined || appId === null) {
        throw new Error("Missing the required parameter 'appId' when calling updateSwitchboard");
      }
      // verify the required parameter 'switchboardId' is set
      if (switchboardId === undefined || switchboardId === null) {
        throw new Error("Missing the required parameter 'switchboardId' when calling updateSwitchboard");
      }
      // verify the required parameter 'switchboardUpdateBody' is set
      if (switchboardUpdateBody === undefined || switchboardUpdateBody === null) {
        throw new Error("Missing the required parameter 'switchboardUpdateBody' when calling updateSwitchboard");
      }
      var pathParams = {
        'appId': appId,
        'switchboardId': switchboardId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['basicAuth', 'bearerAuth'];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = _SwitchboardResponse["default"];
      return this.apiClient.callApi('/v2/apps/{appId}/switchboards/{switchboardId}', 'PATCH', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Update Switchboard
     * Updates a switchboard record.
     * @param {String} appId Identifies the app.
     * @param {String} switchboardId Identifies the switchboard.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardUpdateBody} switchboardUpdateBody 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardResponse}
     */
  }, {
    key: "updateSwitchboard",
    value: function updateSwitchboard(appId, switchboardId, switchboardUpdateBody) {
      return this.updateSwitchboardWithHttpInfo(appId, switchboardId, switchboardUpdateBody).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
  return SwitchboardsApi;
}();
exports["default"] = SwitchboardsApi;
