// sunshine-conversations-client/dist/sunshine-conversations-client/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AcceptControlBody", {
  enumerable: true,
  get: function get() {
    return _AcceptControlBody["default"];
  }
});
Object.defineProperty(exports, "Action", {
  enumerable: true,
  get: function get() {
    return _Action["default"];
  }
});
Object.defineProperty(exports, "ActionSubset", {
  enumerable: true,
  get: function get() {
    return _ActionSubset["default"];
  }
});
Object.defineProperty(exports, "ActivitiesApi", {
  enumerable: true,
  get: function get() {
    return _ActivitiesApi["default"];
  }
});
Object.defineProperty(exports, "Activity", {
  enumerable: true,
  get: function get() {
    return _Activity["default"];
  }
});
Object.defineProperty(exports, "ActivityAllOf", {
  enumerable: true,
  get: function get() {
    return _ActivityAllOf["default"];
  }
});
Object.defineProperty(exports, "ActivityPost", {
  enumerable: true,
  get: function get() {
    return _ActivityPost["default"];
  }
});
Object.defineProperty(exports, "ActivityPostAllOf", {
  enumerable: true,
  get: function get() {
    return _ActivityPostAllOf["default"];
  }
});
Object.defineProperty(exports, "ActivityTypes", {
  enumerable: true,
  get: function get() {
    return _ActivityTypes["default"];
  }
});
Object.defineProperty(exports, "Android", {
  enumerable: true,
  get: function get() {
    return _Android["default"];
  }
});
Object.defineProperty(exports, "AndroidAllOf", {
  enumerable: true,
  get: function get() {
    return _AndroidAllOf["default"];
  }
});
Object.defineProperty(exports, "AndroidUpdate", {
  enumerable: true,
  get: function get() {
    return _AndroidUpdate["default"];
  }
});
Object.defineProperty(exports, "AndroidUpdateAllOf", {
  enumerable: true,
  get: function get() {
    return _AndroidUpdateAllOf["default"];
  }
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "ApiKey", {
  enumerable: true,
  get: function get() {
    return _ApiKey["default"];
  }
});
Object.defineProperty(exports, "App", {
  enumerable: true,
  get: function get() {
    return _App["default"];
  }
});
Object.defineProperty(exports, "AppCreateBody", {
  enumerable: true,
  get: function get() {
    return _AppCreateBody["default"];
  }
});
Object.defineProperty(exports, "AppKey", {
  enumerable: true,
  get: function get() {
    return _AppKey["default"];
  }
});
Object.defineProperty(exports, "AppKeyCreateBody", {
  enumerable: true,
  get: function get() {
    return _AppKeyCreateBody["default"];
  }
});
Object.defineProperty(exports, "AppKeyListResponse", {
  enumerable: true,
  get: function get() {
    return _AppKeyListResponse["default"];
  }
});
Object.defineProperty(exports, "AppKeyResponse", {
  enumerable: true,
  get: function get() {
    return _AppKeyResponse["default"];
  }
});
Object.defineProperty(exports, "AppKeysApi", {
  enumerable: true,
  get: function get() {
    return _AppKeysApi["default"];
  }
});
Object.defineProperty(exports, "AppListFilter", {
  enumerable: true,
  get: function get() {
    return _AppListFilter["default"];
  }
});
Object.defineProperty(exports, "AppListResponse", {
  enumerable: true,
  get: function get() {
    return _AppListResponse["default"];
  }
});
Object.defineProperty(exports, "AppResponse", {
  enumerable: true,
  get: function get() {
    return _AppResponse["default"];
  }
});
Object.defineProperty(exports, "AppSettings", {
  enumerable: true,
  get: function get() {
    return _AppSettings["default"];
  }
});
Object.defineProperty(exports, "AppSubSchema", {
  enumerable: true,
  get: function get() {
    return _AppSubSchema["default"];
  }
});
Object.defineProperty(exports, "AppUpdateBody", {
  enumerable: true,
  get: function get() {
    return _AppUpdateBody["default"];
  }
});
Object.defineProperty(exports, "Apple", {
  enumerable: true,
  get: function get() {
    return _Apple["default"];
  }
});
Object.defineProperty(exports, "AppleAllOf", {
  enumerable: true,
  get: function get() {
    return _AppleAllOf["default"];
  }
});
Object.defineProperty(exports, "AppleUpdate", {
  enumerable: true,
  get: function get() {
    return _AppleUpdate["default"];
  }
});
Object.defineProperty(exports, "AppsApi", {
  enumerable: true,
  get: function get() {
    return _AppsApi["default"];
  }
});
Object.defineProperty(exports, "AttachmentDeleteBody", {
  enumerable: true,
  get: function get() {
    return _AttachmentDeleteBody["default"];
  }
});
Object.defineProperty(exports, "AttachmentMediaTokenBody", {
  enumerable: true,
  get: function get() {
    return _AttachmentMediaTokenBody["default"];
  }
});
Object.defineProperty(exports, "AttachmentMediaTokenResponse", {
  enumerable: true,
  get: function get() {
    return _AttachmentMediaTokenResponse["default"];
  }
});
Object.defineProperty(exports, "AttachmentResponse", {
  enumerable: true,
  get: function get() {
    return _AttachmentResponse["default"];
  }
});
Object.defineProperty(exports, "AttachmentSchema", {
  enumerable: true,
  get: function get() {
    return _AttachmentSchema["default"];
  }
});
Object.defineProperty(exports, "AttachmentUploadBody", {
  enumerable: true,
  get: function get() {
    return _AttachmentUploadBody["default"];
  }
});
Object.defineProperty(exports, "AttachmentsApi", {
  enumerable: true,
  get: function get() {
    return _AttachmentsApi["default"];
  }
});
Object.defineProperty(exports, "Author", {
  enumerable: true,
  get: function get() {
    return _Author["default"];
  }
});
Object.defineProperty(exports, "AuthorWebhook", {
  enumerable: true,
  get: function get() {
    return _AuthorWebhook["default"];
  }
});
Object.defineProperty(exports, "Buy", {
  enumerable: true,
  get: function get() {
    return _Buy["default"];
  }
});
Object.defineProperty(exports, "Campaign", {
  enumerable: true,
  get: function get() {
    return _Campaign["default"];
  }
});
Object.defineProperty(exports, "CarouselMessage", {
  enumerable: true,
  get: function get() {
    return _CarouselMessage["default"];
  }
});
Object.defineProperty(exports, "CarouselMessageDisplaySettings", {
  enumerable: true,
  get: function get() {
    return _CarouselMessageDisplaySettings["default"];
  }
});
Object.defineProperty(exports, "Client", {
  enumerable: true,
  get: function get() {
    return _Client["default"];
  }
});
Object.defineProperty(exports, "ClientAddEvent", {
  enumerable: true,
  get: function get() {
    return _ClientAddEvent["default"];
  }
});
Object.defineProperty(exports, "ClientAddEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ClientAddEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ClientAddEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ClientAddEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ClientAssociation", {
  enumerable: true,
  get: function get() {
    return _ClientAssociation["default"];
  }
});
Object.defineProperty(exports, "ClientCreate", {
  enumerable: true,
  get: function get() {
    return _ClientCreate["default"];
  }
});
Object.defineProperty(exports, "ClientListResponse", {
  enumerable: true,
  get: function get() {
    return _ClientListResponse["default"];
  }
});
Object.defineProperty(exports, "ClientRemoveEvent", {
  enumerable: true,
  get: function get() {
    return _ClientRemoveEvent["default"];
  }
});
Object.defineProperty(exports, "ClientRemoveEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ClientRemoveEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ClientRemoveEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ClientRemoveEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ClientResponse", {
  enumerable: true,
  get: function get() {
    return _ClientResponse["default"];
  }
});
Object.defineProperty(exports, "ClientType", {
  enumerable: true,
  get: function get() {
    return _ClientType["default"];
  }
});
Object.defineProperty(exports, "ClientUpdateEvent", {
  enumerable: true,
  get: function get() {
    return _ClientUpdateEvent["default"];
  }
});
Object.defineProperty(exports, "ClientUpdateEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ClientUpdateEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ClientUpdateEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ClientUpdateEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ClientsApi", {
  enumerable: true,
  get: function get() {
    return _ClientsApi["default"];
  }
});
Object.defineProperty(exports, "Confirmation", {
  enumerable: true,
  get: function get() {
    return _Confirmation["default"];
  }
});
Object.defineProperty(exports, "Content", {
  enumerable: true,
  get: function get() {
    return _Content["default"];
  }
});
Object.defineProperty(exports, "Conversation", {
  enumerable: true,
  get: function get() {
    return _Conversation["default"];
  }
});
Object.defineProperty(exports, "ConversationAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationCreateBody", {
  enumerable: true,
  get: function get() {
    return _ConversationCreateBody["default"];
  }
});
Object.defineProperty(exports, "ConversationCreateEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationCreateEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationCreateEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationCreateEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationCreateEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationCreateEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationJoinEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationJoinEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationJoinEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationJoinEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationJoinEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationJoinEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationLeaveEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationLeaveEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationLeaveEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationLeaveEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationLeaveEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationLeaveEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationListFilter", {
  enumerable: true,
  get: function get() {
    return _ConversationListFilter["default"];
  }
});
Object.defineProperty(exports, "ConversationListResponse", {
  enumerable: true,
  get: function get() {
    return _ConversationListResponse["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageDeliveryChannelEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageDeliveryChannelEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageDeliveryChannelEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageDeliveryChannelEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageDeliveryFailureEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageDeliveryFailureEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageDeliveryFailureEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageDeliveryFailureEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageDeliveryPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageDeliveryPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageDeliveryPayloadDestination", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageDeliveryPayloadDestination["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageDeliveryPayloadExternalMessages", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageDeliveryPayloadExternalMessages["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageDeliveryPayloadMessage", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageDeliveryPayloadMessage["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageDeliveryUserEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageDeliveryUserEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationMessageEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationMessageEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationPostbackEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationPostbackEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationPostbackEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationPostbackEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationPostbackEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationPostbackEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationReadEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationReadEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationReadEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationReadEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationReadEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationReadEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationReferralEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationReferralEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationReferralEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationReferralEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationReferralEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationReferralEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationRemoveEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationRemoveEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationRemoveEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationRemoveEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationRemoveEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationRemoveEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationResponse", {
  enumerable: true,
  get: function get() {
    return _ConversationResponse["default"];
  }
});
Object.defineProperty(exports, "ConversationTruncated", {
  enumerable: true,
  get: function get() {
    return _ConversationTruncated["default"];
  }
});
Object.defineProperty(exports, "ConversationType", {
  enumerable: true,
  get: function get() {
    return _ConversationType["default"];
  }
});
Object.defineProperty(exports, "ConversationTypingEvent", {
  enumerable: true,
  get: function get() {
    return _ConversationTypingEvent["default"];
  }
});
Object.defineProperty(exports, "ConversationTypingEventAllOf", {
  enumerable: true,
  get: function get() {
    return _ConversationTypingEventAllOf["default"];
  }
});
Object.defineProperty(exports, "ConversationTypingEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _ConversationTypingEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "ConversationUpdateBody", {
  enumerable: true,
  get: function get() {
    return _ConversationUpdateBody["default"];
  }
});
Object.defineProperty(exports, "ConversationsApi", {
  enumerable: true,
  get: function get() {
    return _ConversationsApi["default"];
  }
});
Object.defineProperty(exports, "Custom", {
  enumerable: true,
  get: function get() {
    return _Custom["default"];
  }
});
Object.defineProperty(exports, "CustomAllOf", {
  enumerable: true,
  get: function get() {
    return _CustomAllOf["default"];
  }
});
Object.defineProperty(exports, "CustomIntegrationApiKeysApi", {
  enumerable: true,
  get: function get() {
    return _CustomIntegrationApiKeysApi["default"];
  }
});
Object.defineProperty(exports, "CustomUpdate", {
  enumerable: true,
  get: function get() {
    return _CustomUpdate["default"];
  }
});
Object.defineProperty(exports, "Destination", {
  enumerable: true,
  get: function get() {
    return _Destination["default"];
  }
});
Object.defineProperty(exports, "Device", {
  enumerable: true,
  get: function get() {
    return _Device["default"];
  }
});
Object.defineProperty(exports, "EventSubSchema", {
  enumerable: true,
  get: function get() {
    return _EventSubSchema["default"];
  }
});
Object.defineProperty(exports, "ExtraChannelOptions", {
  enumerable: true,
  get: function get() {
    return _ExtraChannelOptions["default"];
  }
});
Object.defineProperty(exports, "ExtraChannelOptionsMessenger", {
  enumerable: true,
  get: function get() {
    return _ExtraChannelOptionsMessenger["default"];
  }
});
Object.defineProperty(exports, "Field", {
  enumerable: true,
  get: function get() {
    return _Field["default"];
  }
});
Object.defineProperty(exports, "FileMessage", {
  enumerable: true,
  get: function get() {
    return _FileMessage["default"];
  }
});
Object.defineProperty(exports, "FormMessage", {
  enumerable: true,
  get: function get() {
    return _FormMessage["default"];
  }
});
Object.defineProperty(exports, "FormResponseMessage", {
  enumerable: true,
  get: function get() {
    return _FormResponseMessage["default"];
  }
});
Object.defineProperty(exports, "Identity", {
  enumerable: true,
  get: function get() {
    return _Identity["default"];
  }
});
Object.defineProperty(exports, "ImageMessage", {
  enumerable: true,
  get: function get() {
    return _ImageMessage["default"];
  }
});
Object.defineProperty(exports, "InlineObject", {
  enumerable: true,
  get: function get() {
    return _InlineObject["default"];
  }
});
Object.defineProperty(exports, "Instagram", {
  enumerable: true,
  get: function get() {
    return _Instagram["default"];
  }
});
Object.defineProperty(exports, "InstagramAllOf", {
  enumerable: true,
  get: function get() {
    return _InstagramAllOf["default"];
  }
});
Object.defineProperty(exports, "InstagramUpdate", {
  enumerable: true,
  get: function get() {
    return _InstagramUpdate["default"];
  }
});
Object.defineProperty(exports, "InstagramUpdateAllOf", {
  enumerable: true,
  get: function get() {
    return _InstagramUpdateAllOf["default"];
  }
});
Object.defineProperty(exports, "Integration", {
  enumerable: true,
  get: function get() {
    return _Integration["default"];
  }
});
Object.defineProperty(exports, "IntegrationApiKey", {
  enumerable: true,
  get: function get() {
    return _IntegrationApiKey["default"];
  }
});
Object.defineProperty(exports, "IntegrationApiKeyListResponse", {
  enumerable: true,
  get: function get() {
    return _IntegrationApiKeyListResponse["default"];
  }
});
Object.defineProperty(exports, "IntegrationApiKeyResponse", {
  enumerable: true,
  get: function get() {
    return _IntegrationApiKeyResponse["default"];
  }
});
Object.defineProperty(exports, "IntegrationId", {
  enumerable: true,
  get: function get() {
    return _IntegrationId["default"];
  }
});
Object.defineProperty(exports, "IntegrationListFilter", {
  enumerable: true,
  get: function get() {
    return _IntegrationListFilter["default"];
  }
});
Object.defineProperty(exports, "IntegrationListResponse", {
  enumerable: true,
  get: function get() {
    return _IntegrationListResponse["default"];
  }
});
Object.defineProperty(exports, "IntegrationResponse", {
  enumerable: true,
  get: function get() {
    return _IntegrationResponse["default"];
  }
});
Object.defineProperty(exports, "IntegrationType", {
  enumerable: true,
  get: function get() {
    return _IntegrationType["default"];
  }
});
Object.defineProperty(exports, "IntegrationUpdate", {
  enumerable: true,
  get: function get() {
    return _IntegrationUpdate["default"];
  }
});
Object.defineProperty(exports, "IntegrationUpdateBase", {
  enumerable: true,
  get: function get() {
    return _IntegrationUpdateBase["default"];
  }
});
Object.defineProperty(exports, "IntegrationsApi", {
  enumerable: true,
  get: function get() {
    return _IntegrationsApi["default"];
  }
});
Object.defineProperty(exports, "Ios", {
  enumerable: true,
  get: function get() {
    return _Ios["default"];
  }
});
Object.defineProperty(exports, "IosAllOf", {
  enumerable: true,
  get: function get() {
    return _IosAllOf["default"];
  }
});
Object.defineProperty(exports, "IosUpdate", {
  enumerable: true,
  get: function get() {
    return _IosUpdate["default"];
  }
});
Object.defineProperty(exports, "IosUpdateAllOf", {
  enumerable: true,
  get: function get() {
    return _IosUpdateAllOf["default"];
  }
});
Object.defineProperty(exports, "Item", {
  enumerable: true,
  get: function get() {
    return _Item["default"];
  }
});
Object.defineProperty(exports, "Line", {
  enumerable: true,
  get: function get() {
    return _Line["default"];
  }
});
Object.defineProperty(exports, "LineAllOf", {
  enumerable: true,
  get: function get() {
    return _LineAllOf["default"];
  }
});
Object.defineProperty(exports, "LineUpdate", {
  enumerable: true,
  get: function get() {
    return _LineUpdate["default"];
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _Link["default"];
  }
});
Object.defineProperty(exports, "Links", {
  enumerable: true,
  get: function get() {
    return _Links["default"];
  }
});
Object.defineProperty(exports, "ListMessage", {
  enumerable: true,
  get: function get() {
    return _ListMessage["default"];
  }
});
Object.defineProperty(exports, "LocationMessage", {
  enumerable: true,
  get: function get() {
    return _LocationMessage["default"];
  }
});
Object.defineProperty(exports, "LocationMessageCoordinates", {
  enumerable: true,
  get: function get() {
    return _LocationMessageCoordinates["default"];
  }
});
Object.defineProperty(exports, "LocationMessageLocation", {
  enumerable: true,
  get: function get() {
    return _LocationMessageLocation["default"];
  }
});
Object.defineProperty(exports, "LocationRequest", {
  enumerable: true,
  get: function get() {
    return _LocationRequest["default"];
  }
});
Object.defineProperty(exports, "Mailgun", {
  enumerable: true,
  get: function get() {
    return _Mailgun["default"];
  }
});
Object.defineProperty(exports, "MailgunAllOf", {
  enumerable: true,
  get: function get() {
    return _MailgunAllOf["default"];
  }
});
Object.defineProperty(exports, "MailgunUpdate", {
  enumerable: true,
  get: function get() {
    return _MailgunUpdate["default"];
  }
});
Object.defineProperty(exports, "MailgunUpdateAllOf", {
  enumerable: true,
  get: function get() {
    return _MailgunUpdateAllOf["default"];
  }
});
Object.defineProperty(exports, "MatchCriteria", {
  enumerable: true,
  get: function get() {
    return _MatchCriteria["default"];
  }
});
Object.defineProperty(exports, "MatchCriteriaBase", {
  enumerable: true,
  get: function get() {
    return _MatchCriteriaBase["default"];
  }
});
Object.defineProperty(exports, "MatchCriteriaMailgun", {
  enumerable: true,
  get: function get() {
    return _MatchCriteriaMailgun["default"];
  }
});
Object.defineProperty(exports, "MatchCriteriaMailgunAllOf", {
  enumerable: true,
  get: function get() {
    return _MatchCriteriaMailgunAllOf["default"];
  }
});
Object.defineProperty(exports, "MatchCriteriaMessagebird", {
  enumerable: true,
  get: function get() {
    return _MatchCriteriaMessagebird["default"];
  }
});
Object.defineProperty(exports, "MatchCriteriaMessagebirdAllOf", {
  enumerable: true,
  get: function get() {
    return _MatchCriteriaMessagebirdAllOf["default"];
  }
});
Object.defineProperty(exports, "MatchCriteriaTwilio", {
  enumerable: true,
  get: function get() {
    return _MatchCriteriaTwilio["default"];
  }
});
Object.defineProperty(exports, "MatchCriteriaTwilioAllOf", {
  enumerable: true,
  get: function get() {
    return _MatchCriteriaTwilioAllOf["default"];
  }
});
Object.defineProperty(exports, "MatchCriteriaWhatsapp", {
  enumerable: true,
  get: function get() {
    return _MatchCriteriaWhatsapp["default"];
  }
});
Object.defineProperty(exports, "MatchCriteriaWhatsappAllOf", {
  enumerable: true,
  get: function get() {
    return _MatchCriteriaWhatsappAllOf["default"];
  }
});
Object.defineProperty(exports, "Message", {
  enumerable: true,
  get: function get() {
    return _Message["default"];
  }
});
Object.defineProperty(exports, "MessageBirdUpdate", {
  enumerable: true,
  get: function get() {
    return _MessageBirdUpdate["default"];
  }
});
Object.defineProperty(exports, "MessageListResponse", {
  enumerable: true,
  get: function get() {
    return _MessageListResponse["default"];
  }
});
Object.defineProperty(exports, "MessageOverride", {
  enumerable: true,
  get: function get() {
    return _MessageOverride["default"];
  }
});
Object.defineProperty(exports, "MessageOverrideApple", {
  enumerable: true,
  get: function get() {
    return _MessageOverrideApple["default"];
  }
});
Object.defineProperty(exports, "MessageOverrideLine", {
  enumerable: true,
  get: function get() {
    return _MessageOverrideLine["default"];
  }
});
Object.defineProperty(exports, "MessageOverrideMessenger", {
  enumerable: true,
  get: function get() {
    return _MessageOverrideMessenger["default"];
  }
});
Object.defineProperty(exports, "MessageOverridePayload", {
  enumerable: true,
  get: function get() {
    return _MessageOverridePayload["default"];
  }
});
Object.defineProperty(exports, "MessageOverrideWhatsapp", {
  enumerable: true,
  get: function get() {
    return _MessageOverrideWhatsapp["default"];
  }
});
Object.defineProperty(exports, "MessagePost", {
  enumerable: true,
  get: function get() {
    return _MessagePost["default"];
  }
});
Object.defineProperty(exports, "MessagePostResponse", {
  enumerable: true,
  get: function get() {
    return _MessagePostResponse["default"];
  }
});
Object.defineProperty(exports, "MessageWebhook", {
  enumerable: true,
  get: function get() {
    return _MessageWebhook["default"];
  }
});
Object.defineProperty(exports, "Messagebird", {
  enumerable: true,
  get: function get() {
    return _Messagebird["default"];
  }
});
Object.defineProperty(exports, "MessagebirdAllOf", {
  enumerable: true,
  get: function get() {
    return _MessagebirdAllOf["default"];
  }
});
Object.defineProperty(exports, "MessagesApi", {
  enumerable: true,
  get: function get() {
    return _MessagesApi["default"];
  }
});
Object.defineProperty(exports, "Messenger", {
  enumerable: true,
  get: function get() {
    return _Messenger["default"];
  }
});
Object.defineProperty(exports, "MessengerAllOf", {
  enumerable: true,
  get: function get() {
    return _MessengerAllOf["default"];
  }
});
Object.defineProperty(exports, "MessengerUpdate", {
  enumerable: true,
  get: function get() {
    return _MessengerUpdate["default"];
  }
});
Object.defineProperty(exports, "Meta", {
  enumerable: true,
  get: function get() {
    return _Meta["default"];
  }
});
Object.defineProperty(exports, "OAuthEndpointsApi", {
  enumerable: true,
  get: function get() {
    return _OAuthEndpointsApi["default"];
  }
});
Object.defineProperty(exports, "OfferControlBody", {
  enumerable: true,
  get: function get() {
    return _OfferControlBody["default"];
  }
});
Object.defineProperty(exports, "Page", {
  enumerable: true,
  get: function get() {
    return _Page["default"];
  }
});
Object.defineProperty(exports, "Participant", {
  enumerable: true,
  get: function get() {
    return _Participant["default"];
  }
});
Object.defineProperty(exports, "ParticipantJoinBody", {
  enumerable: true,
  get: function get() {
    return _ParticipantJoinBody["default"];
  }
});
Object.defineProperty(exports, "ParticipantLeaveBody", {
  enumerable: true,
  get: function get() {
    return _ParticipantLeaveBody["default"];
  }
});
Object.defineProperty(exports, "ParticipantLeaveBodyParticipantId", {
  enumerable: true,
  get: function get() {
    return _ParticipantLeaveBodyParticipantId["default"];
  }
});
Object.defineProperty(exports, "ParticipantLeaveBodyUserExternalId", {
  enumerable: true,
  get: function get() {
    return _ParticipantLeaveBodyUserExternalId["default"];
  }
});
Object.defineProperty(exports, "ParticipantLeaveBodyUserId", {
  enumerable: true,
  get: function get() {
    return _ParticipantLeaveBodyUserId["default"];
  }
});
Object.defineProperty(exports, "ParticipantListResponse", {
  enumerable: true,
  get: function get() {
    return _ParticipantListResponse["default"];
  }
});
Object.defineProperty(exports, "ParticipantResponse", {
  enumerable: true,
  get: function get() {
    return _ParticipantResponse["default"];
  }
});
Object.defineProperty(exports, "ParticipantSubSchema", {
  enumerable: true,
  get: function get() {
    return _ParticipantSubSchema["default"];
  }
});
Object.defineProperty(exports, "ParticipantWithUserExternalId", {
  enumerable: true,
  get: function get() {
    return _ParticipantWithUserExternalId["default"];
  }
});
Object.defineProperty(exports, "ParticipantWithUserId", {
  enumerable: true,
  get: function get() {
    return _ParticipantWithUserId["default"];
  }
});
Object.defineProperty(exports, "ParticipantsApi", {
  enumerable: true,
  get: function get() {
    return _ParticipantsApi["default"];
  }
});
Object.defineProperty(exports, "PassControlBody", {
  enumerable: true,
  get: function get() {
    return _PassControlBody["default"];
  }
});
Object.defineProperty(exports, "Postback", {
  enumerable: true,
  get: function get() {
    return _Postback["default"];
  }
});
Object.defineProperty(exports, "PostbackWebhook", {
  enumerable: true,
  get: function get() {
    return _PostbackWebhook["default"];
  }
});
Object.defineProperty(exports, "PrechatCapture", {
  enumerable: true,
  get: function get() {
    return _PrechatCapture["default"];
  }
});
Object.defineProperty(exports, "Profile", {
  enumerable: true,
  get: function get() {
    return _Profile["default"];
  }
});
Object.defineProperty(exports, "QuotedMessage", {
  enumerable: true,
  get: function get() {
    return _QuotedMessage["default"];
  }
});
Object.defineProperty(exports, "QuotedMessageExternalMessageId", {
  enumerable: true,
  get: function get() {
    return _QuotedMessageExternalMessageId["default"];
  }
});
Object.defineProperty(exports, "QuotedMessageMessage", {
  enumerable: true,
  get: function get() {
    return _QuotedMessageMessage["default"];
  }
});
Object.defineProperty(exports, "Referral", {
  enumerable: true,
  get: function get() {
    return _Referral["default"];
  }
});
Object.defineProperty(exports, "ReferralDetails", {
  enumerable: true,
  get: function get() {
    return _ReferralDetails["default"];
  }
});
Object.defineProperty(exports, "Reply", {
  enumerable: true,
  get: function get() {
    return _Reply["default"];
  }
});
Object.defineProperty(exports, "Source", {
  enumerable: true,
  get: function get() {
    return _Source["default"];
  }
});
Object.defineProperty(exports, "SourceWebhook", {
  enumerable: true,
  get: function get() {
    return _SourceWebhook["default"];
  }
});
Object.defineProperty(exports, "SourceWithCampaignWebhook", {
  enumerable: true,
  get: function get() {
    return _SourceWithCampaignWebhook["default"];
  }
});
Object.defineProperty(exports, "SourceWithCampaignWebhookAllOf", {
  enumerable: true,
  get: function get() {
    return _SourceWithCampaignWebhookAllOf["default"];
  }
});
Object.defineProperty(exports, "Status", {
  enumerable: true,
  get: function get() {
    return _Status["default"];
  }
});
Object.defineProperty(exports, "Switchboard", {
  enumerable: true,
  get: function get() {
    return _Switchboard["default"];
  }
});
Object.defineProperty(exports, "SwitchboardAcceptControl", {
  enumerable: true,
  get: function get() {
    return _SwitchboardAcceptControl["default"];
  }
});
Object.defineProperty(exports, "SwitchboardAcceptControlAllOf", {
  enumerable: true,
  get: function get() {
    return _SwitchboardAcceptControlAllOf["default"];
  }
});
Object.defineProperty(exports, "SwitchboardAcceptControlAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _SwitchboardAcceptControlAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "SwitchboardAcceptControlFailure", {
  enumerable: true,
  get: function get() {
    return _SwitchboardAcceptControlFailure["default"];
  }
});
Object.defineProperty(exports, "SwitchboardAcceptControlFailureAllOf", {
  enumerable: true,
  get: function get() {
    return _SwitchboardAcceptControlFailureAllOf["default"];
  }
});
Object.defineProperty(exports, "SwitchboardAcceptControlFailureAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _SwitchboardAcceptControlFailureAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "SwitchboardActionsApi", {
  enumerable: true,
  get: function get() {
    return _SwitchboardActionsApi["default"];
  }
});
Object.defineProperty(exports, "SwitchboardIntegration", {
  enumerable: true,
  get: function get() {
    return _SwitchboardIntegration["default"];
  }
});
Object.defineProperty(exports, "SwitchboardIntegrationCreateBody", {
  enumerable: true,
  get: function get() {
    return _SwitchboardIntegrationCreateBody["default"];
  }
});
Object.defineProperty(exports, "SwitchboardIntegrationListResponse", {
  enumerable: true,
  get: function get() {
    return _SwitchboardIntegrationListResponse["default"];
  }
});
Object.defineProperty(exports, "SwitchboardIntegrationResponse", {
  enumerable: true,
  get: function get() {
    return _SwitchboardIntegrationResponse["default"];
  }
});
Object.defineProperty(exports, "SwitchboardIntegrationUpdateBody", {
  enumerable: true,
  get: function get() {
    return _SwitchboardIntegrationUpdateBody["default"];
  }
});
Object.defineProperty(exports, "SwitchboardIntegrationWebhook", {
  enumerable: true,
  get: function get() {
    return _SwitchboardIntegrationWebhook["default"];
  }
});
Object.defineProperty(exports, "SwitchboardIntegrationsApi", {
  enumerable: true,
  get: function get() {
    return _SwitchboardIntegrationsApi["default"];
  }
});
Object.defineProperty(exports, "SwitchboardListResponse", {
  enumerable: true,
  get: function get() {
    return _SwitchboardListResponse["default"];
  }
});
Object.defineProperty(exports, "SwitchboardOfferControl", {
  enumerable: true,
  get: function get() {
    return _SwitchboardOfferControl["default"];
  }
});
Object.defineProperty(exports, "SwitchboardOfferControlAllOf", {
  enumerable: true,
  get: function get() {
    return _SwitchboardOfferControlAllOf["default"];
  }
});
Object.defineProperty(exports, "SwitchboardOfferControlAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _SwitchboardOfferControlAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "SwitchboardOfferControlFailure", {
  enumerable: true,
  get: function get() {
    return _SwitchboardOfferControlFailure["default"];
  }
});
Object.defineProperty(exports, "SwitchboardPassControl", {
  enumerable: true,
  get: function get() {
    return _SwitchboardPassControl["default"];
  }
});
Object.defineProperty(exports, "SwitchboardPassControlAllOf", {
  enumerable: true,
  get: function get() {
    return _SwitchboardPassControlAllOf["default"];
  }
});
Object.defineProperty(exports, "SwitchboardPassControlAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _SwitchboardPassControlAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "SwitchboardPassControlFailure", {
  enumerable: true,
  get: function get() {
    return _SwitchboardPassControlFailure["default"];
  }
});
Object.defineProperty(exports, "SwitchboardResponse", {
  enumerable: true,
  get: function get() {
    return _SwitchboardResponse["default"];
  }
});
Object.defineProperty(exports, "SwitchboardUpdateBody", {
  enumerable: true,
  get: function get() {
    return _SwitchboardUpdateBody["default"];
  }
});
Object.defineProperty(exports, "SwitchboardsApi", {
  enumerable: true,
  get: function get() {
    return _SwitchboardsApi["default"];
  }
});
Object.defineProperty(exports, "Target", {
  enumerable: true,
  get: function get() {
    return _Target["default"];
  }
});
Object.defineProperty(exports, "Telegram", {
  enumerable: true,
  get: function get() {
    return _Telegram["default"];
  }
});
Object.defineProperty(exports, "TelegramAllOf", {
  enumerable: true,
  get: function get() {
    return _TelegramAllOf["default"];
  }
});
Object.defineProperty(exports, "TelegramUpdate", {
  enumerable: true,
  get: function get() {
    return _TelegramUpdate["default"];
  }
});
Object.defineProperty(exports, "TemplateMessage", {
  enumerable: true,
  get: function get() {
    return _TemplateMessage["default"];
  }
});
Object.defineProperty(exports, "TextMessage", {
  enumerable: true,
  get: function get() {
    return _TextMessage["default"];
  }
});
Object.defineProperty(exports, "Twilio", {
  enumerable: true,
  get: function get() {
    return _Twilio["default"];
  }
});
Object.defineProperty(exports, "TwilioAllOf", {
  enumerable: true,
  get: function get() {
    return _TwilioAllOf["default"];
  }
});
Object.defineProperty(exports, "TwilioUpdate", {
  enumerable: true,
  get: function get() {
    return _TwilioUpdate["default"];
  }
});
Object.defineProperty(exports, "Twitter", {
  enumerable: true,
  get: function get() {
    return _Twitter["default"];
  }
});
Object.defineProperty(exports, "TwitterAllOf", {
  enumerable: true,
  get: function get() {
    return _TwitterAllOf["default"];
  }
});
Object.defineProperty(exports, "TwitterUpdate", {
  enumerable: true,
  get: function get() {
    return _TwitterUpdate["default"];
  }
});
Object.defineProperty(exports, "User", {
  enumerable: true,
  get: function get() {
    return _User["default"];
  }
});
Object.defineProperty(exports, "UserAllOf", {
  enumerable: true,
  get: function get() {
    return _UserAllOf["default"];
  }
});
Object.defineProperty(exports, "UserCreateBody", {
  enumerable: true,
  get: function get() {
    return _UserCreateBody["default"];
  }
});
Object.defineProperty(exports, "UserMergeEvent", {
  enumerable: true,
  get: function get() {
    return _UserMergeEvent["default"];
  }
});
Object.defineProperty(exports, "UserMergeEventAllOf", {
  enumerable: true,
  get: function get() {
    return _UserMergeEventAllOf["default"];
  }
});
Object.defineProperty(exports, "UserMergeEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _UserMergeEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "UserMergeEventAllOfPayloadMergedClients", {
  enumerable: true,
  get: function get() {
    return _UserMergeEventAllOfPayloadMergedClients["default"];
  }
});
Object.defineProperty(exports, "UserMergeEventAllOfPayloadMergedConversations", {
  enumerable: true,
  get: function get() {
    return _UserMergeEventAllOfPayloadMergedConversations["default"];
  }
});
Object.defineProperty(exports, "UserMergeEventAllOfPayloadMergedUsers", {
  enumerable: true,
  get: function get() {
    return _UserMergeEventAllOfPayloadMergedUsers["default"];
  }
});
Object.defineProperty(exports, "UserResponse", {
  enumerable: true,
  get: function get() {
    return _UserResponse["default"];
  }
});
Object.defineProperty(exports, "UserTruncated", {
  enumerable: true,
  get: function get() {
    return _UserTruncated["default"];
  }
});
Object.defineProperty(exports, "UserUpdateBody", {
  enumerable: true,
  get: function get() {
    return _UserUpdateBody["default"];
  }
});
Object.defineProperty(exports, "UserUpdateEvent", {
  enumerable: true,
  get: function get() {
    return _UserUpdateEvent["default"];
  }
});
Object.defineProperty(exports, "UserUpdateEventAllOf", {
  enumerable: true,
  get: function get() {
    return _UserUpdateEventAllOf["default"];
  }
});
Object.defineProperty(exports, "UserUpdateEventAllOfPayload", {
  enumerable: true,
  get: function get() {
    return _UserUpdateEventAllOfPayload["default"];
  }
});
Object.defineProperty(exports, "UsersApi", {
  enumerable: true,
  get: function get() {
    return _UsersApi["default"];
  }
});
Object.defineProperty(exports, "Viber", {
  enumerable: true,
  get: function get() {
    return _Viber["default"];
  }
});
Object.defineProperty(exports, "ViberAllOf", {
  enumerable: true,
  get: function get() {
    return _ViberAllOf["default"];
  }
});
Object.defineProperty(exports, "ViberUpdate", {
  enumerable: true,
  get: function get() {
    return _ViberUpdate["default"];
  }
});
Object.defineProperty(exports, "Web", {
  enumerable: true,
  get: function get() {
    return _Web["default"];
  }
});
Object.defineProperty(exports, "WebAllOf", {
  enumerable: true,
  get: function get() {
    return _WebAllOf["default"];
  }
});
Object.defineProperty(exports, "WebUpdate", {
  enumerable: true,
  get: function get() {
    return _WebUpdate["default"];
  }
});
Object.defineProperty(exports, "WebUpdateAllOf", {
  enumerable: true,
  get: function get() {
    return _WebUpdateAllOf["default"];
  }
});
Object.defineProperty(exports, "Webhook", {
  enumerable: true,
  get: function get() {
    return _Webhook["default"];
  }
});
Object.defineProperty(exports, "WebhookBody", {
  enumerable: true,
  get: function get() {
    return _WebhookBody["default"];
  }
});
Object.defineProperty(exports, "WebhookCreateBody", {
  enumerable: true,
  get: function get() {
    return _WebhookCreateBody["default"];
  }
});
Object.defineProperty(exports, "WebhookListResponse", {
  enumerable: true,
  get: function get() {
    return _WebhookListResponse["default"];
  }
});
Object.defineProperty(exports, "WebhookResponse", {
  enumerable: true,
  get: function get() {
    return _WebhookResponse["default"];
  }
});
Object.defineProperty(exports, "WebhookSubSchema", {
  enumerable: true,
  get: function get() {
    return _WebhookSubSchema["default"];
  }
});
Object.defineProperty(exports, "WebhooksApi", {
  enumerable: true,
  get: function get() {
    return _WebhooksApi["default"];
  }
});
Object.defineProperty(exports, "Webview", {
  enumerable: true,
  get: function get() {
    return _Webview["default"];
  }
});
Object.defineProperty(exports, "WhatsAppUpdate", {
  enumerable: true,
  get: function get() {
    return _WhatsAppUpdate["default"];
  }
});
Object.defineProperty(exports, "WhatsAppUpdateAllOf", {
  enumerable: true,
  get: function get() {
    return _WhatsAppUpdateAllOf["default"];
  }
});
Object.defineProperty(exports, "Whatsapp", {
  enumerable: true,
  get: function get() {
    return _Whatsapp["default"];
  }
});
Object.defineProperty(exports, "WhatsappAllOf", {
  enumerable: true,
  get: function get() {
    return _WhatsappAllOf["default"];
  }
});
var _ApiClient = _interopRequireDefault(require("./ApiClient"));
var _AcceptControlBody = _interopRequireDefault(require("./sunshine-conversations-client.model/AcceptControlBody"));
var _Action = _interopRequireDefault(require("./sunshine-conversations-client.model/Action"));
var _ActionSubset = _interopRequireDefault(require("./sunshine-conversations-client.model/ActionSubset"));
var _Activity = _interopRequireDefault(require("./sunshine-conversations-client.model/Activity"));
var _ActivityAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ActivityAllOf"));
var _ActivityPost = _interopRequireDefault(require("./sunshine-conversations-client.model/ActivityPost"));
var _ActivityPostAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ActivityPostAllOf"));
var _ActivityTypes = _interopRequireDefault(require("./sunshine-conversations-client.model/ActivityTypes"));
var _Android = _interopRequireDefault(require("./sunshine-conversations-client.model/Android"));
var _AndroidAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/AndroidAllOf"));
var _AndroidUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/AndroidUpdate"));
var _AndroidUpdateAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/AndroidUpdateAllOf"));
var _ApiKey = _interopRequireDefault(require("./sunshine-conversations-client.model/ApiKey"));
var _App = _interopRequireDefault(require("./sunshine-conversations-client.model/App"));
var _AppCreateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/AppCreateBody"));
var _AppKey = _interopRequireDefault(require("./sunshine-conversations-client.model/AppKey"));
var _AppKeyCreateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/AppKeyCreateBody"));
var _AppKeyListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/AppKeyListResponse"));
var _AppKeyResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/AppKeyResponse"));
var _AppListFilter = _interopRequireDefault(require("./sunshine-conversations-client.model/AppListFilter"));
var _AppListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/AppListResponse"));
var _AppResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/AppResponse"));
var _AppSettings = _interopRequireDefault(require("./sunshine-conversations-client.model/AppSettings"));
var _AppSubSchema = _interopRequireDefault(require("./sunshine-conversations-client.model/AppSubSchema"));
var _AppUpdateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/AppUpdateBody"));
var _Apple = _interopRequireDefault(require("./sunshine-conversations-client.model/Apple"));
var _AppleAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/AppleAllOf"));
var _AppleUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/AppleUpdate"));
var _AttachmentDeleteBody = _interopRequireDefault(require("./sunshine-conversations-client.model/AttachmentDeleteBody"));
var _AttachmentMediaTokenBody = _interopRequireDefault(require("./sunshine-conversations-client.model/AttachmentMediaTokenBody"));
var _AttachmentMediaTokenResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/AttachmentMediaTokenResponse"));
var _AttachmentResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/AttachmentResponse"));
var _AttachmentSchema = _interopRequireDefault(require("./sunshine-conversations-client.model/AttachmentSchema"));
var _AttachmentUploadBody = _interopRequireDefault(require("./sunshine-conversations-client.model/AttachmentUploadBody"));
var _Author = _interopRequireDefault(require("./sunshine-conversations-client.model/Author"));
var _AuthorWebhook = _interopRequireDefault(require("./sunshine-conversations-client.model/AuthorWebhook"));
var _Buy = _interopRequireDefault(require("./sunshine-conversations-client.model/Buy"));
var _Campaign = _interopRequireDefault(require("./sunshine-conversations-client.model/Campaign"));
var _CarouselMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/CarouselMessage"));
var _CarouselMessageDisplaySettings = _interopRequireDefault(require("./sunshine-conversations-client.model/CarouselMessageDisplaySettings"));
var _Client = _interopRequireDefault(require("./sunshine-conversations-client.model/Client"));
var _ClientAddEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientAddEvent"));
var _ClientAddEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientAddEventAllOf"));
var _ClientAddEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientAddEventAllOfPayload"));
var _ClientAssociation = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientAssociation"));
var _ClientCreate = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientCreate"));
var _ClientListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientListResponse"));
var _ClientRemoveEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientRemoveEvent"));
var _ClientRemoveEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientRemoveEventAllOf"));
var _ClientRemoveEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientRemoveEventAllOfPayload"));
var _ClientResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientResponse"));
var _ClientType = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientType"));
var _ClientUpdateEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientUpdateEvent"));
var _ClientUpdateEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientUpdateEventAllOf"));
var _ClientUpdateEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ClientUpdateEventAllOfPayload"));
var _Confirmation = _interopRequireDefault(require("./sunshine-conversations-client.model/Confirmation"));
var _Content = _interopRequireDefault(require("./sunshine-conversations-client.model/Content"));
var _Conversation = _interopRequireDefault(require("./sunshine-conversations-client.model/Conversation"));
var _ConversationAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationAllOf"));
var _ConversationCreateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationCreateBody"));
var _ConversationCreateEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationCreateEvent"));
var _ConversationCreateEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationCreateEventAllOf"));
var _ConversationCreateEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationCreateEventAllOfPayload"));
var _ConversationJoinEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationJoinEvent"));
var _ConversationJoinEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationJoinEventAllOf"));
var _ConversationJoinEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationJoinEventAllOfPayload"));
var _ConversationLeaveEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationLeaveEvent"));
var _ConversationLeaveEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationLeaveEventAllOf"));
var _ConversationLeaveEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationLeaveEventAllOfPayload"));
var _ConversationListFilter = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationListFilter"));
var _ConversationListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationListResponse"));
var _ConversationMessageDeliveryChannelEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageDeliveryChannelEvent"));
var _ConversationMessageDeliveryChannelEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageDeliveryChannelEventAllOf"));
var _ConversationMessageDeliveryFailureEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageDeliveryFailureEvent"));
var _ConversationMessageDeliveryFailureEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageDeliveryFailureEventAllOf"));
var _ConversationMessageDeliveryPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageDeliveryPayload"));
var _ConversationMessageDeliveryPayloadDestination = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageDeliveryPayloadDestination"));
var _ConversationMessageDeliveryPayloadExternalMessages = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageDeliveryPayloadExternalMessages"));
var _ConversationMessageDeliveryPayloadMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageDeliveryPayloadMessage"));
var _ConversationMessageDeliveryUserEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageDeliveryUserEvent"));
var _ConversationMessageEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageEvent"));
var _ConversationMessageEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageEventAllOf"));
var _ConversationMessageEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationMessageEventAllOfPayload"));
var _ConversationPostbackEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationPostbackEvent"));
var _ConversationPostbackEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationPostbackEventAllOf"));
var _ConversationPostbackEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationPostbackEventAllOfPayload"));
var _ConversationReadEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationReadEvent"));
var _ConversationReadEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationReadEventAllOf"));
var _ConversationReadEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationReadEventAllOfPayload"));
var _ConversationReferralEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationReferralEvent"));
var _ConversationReferralEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationReferralEventAllOf"));
var _ConversationReferralEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationReferralEventAllOfPayload"));
var _ConversationRemoveEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationRemoveEvent"));
var _ConversationRemoveEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationRemoveEventAllOf"));
var _ConversationRemoveEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationRemoveEventAllOfPayload"));
var _ConversationResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationResponse"));
var _ConversationTruncated = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationTruncated"));
var _ConversationType = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationType"));
var _ConversationTypingEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationTypingEvent"));
var _ConversationTypingEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationTypingEventAllOf"));
var _ConversationTypingEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationTypingEventAllOfPayload"));
var _ConversationUpdateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/ConversationUpdateBody"));
var _Custom = _interopRequireDefault(require("./sunshine-conversations-client.model/Custom"));
var _CustomAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/CustomAllOf"));
var _CustomUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/CustomUpdate"));
var _Destination = _interopRequireDefault(require("./sunshine-conversations-client.model/Destination"));
var _Device = _interopRequireDefault(require("./sunshine-conversations-client.model/Device"));
var _EventSubSchema = _interopRequireDefault(require("./sunshine-conversations-client.model/EventSubSchema"));
var _ExtraChannelOptions = _interopRequireDefault(require("./sunshine-conversations-client.model/ExtraChannelOptions"));
var _ExtraChannelOptionsMessenger = _interopRequireDefault(require("./sunshine-conversations-client.model/ExtraChannelOptionsMessenger"));
var _Field = _interopRequireDefault(require("./sunshine-conversations-client.model/Field"));
var _FileMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/FileMessage"));
var _FormMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/FormMessage"));
var _FormResponseMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/FormResponseMessage"));
var _Identity = _interopRequireDefault(require("./sunshine-conversations-client.model/Identity"));
var _ImageMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/ImageMessage"));
var _InlineObject = _interopRequireDefault(require("./sunshine-conversations-client.model/InlineObject"));
var _Instagram = _interopRequireDefault(require("./sunshine-conversations-client.model/Instagram"));
var _InstagramAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/InstagramAllOf"));
var _InstagramUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/InstagramUpdate"));
var _InstagramUpdateAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/InstagramUpdateAllOf"));
var _Integration = _interopRequireDefault(require("./sunshine-conversations-client.model/Integration"));
var _IntegrationApiKey = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationApiKey"));
var _IntegrationApiKeyListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationApiKeyListResponse"));
var _IntegrationApiKeyResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationApiKeyResponse"));
var _IntegrationId = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationId"));
var _IntegrationListFilter = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationListFilter"));
var _IntegrationListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationListResponse"));
var _IntegrationResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationResponse"));
var _IntegrationType = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationType"));
var _IntegrationUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationUpdate"));
var _IntegrationUpdateBase = _interopRequireDefault(require("./sunshine-conversations-client.model/IntegrationUpdateBase"));
var _Ios = _interopRequireDefault(require("./sunshine-conversations-client.model/Ios"));
var _IosAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/IosAllOf"));
var _IosUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/IosUpdate"));
var _IosUpdateAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/IosUpdateAllOf"));
var _Item = _interopRequireDefault(require("./sunshine-conversations-client.model/Item"));
var _Line = _interopRequireDefault(require("./sunshine-conversations-client.model/Line"));
var _LineAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/LineAllOf"));
var _LineUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/LineUpdate"));
var _Link = _interopRequireDefault(require("./sunshine-conversations-client.model/Link"));
var _Links = _interopRequireDefault(require("./sunshine-conversations-client.model/Links"));
var _ListMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/ListMessage"));
var _LocationMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/LocationMessage"));
var _LocationMessageCoordinates = _interopRequireDefault(require("./sunshine-conversations-client.model/LocationMessageCoordinates"));
var _LocationMessageLocation = _interopRequireDefault(require("./sunshine-conversations-client.model/LocationMessageLocation"));
var _LocationRequest = _interopRequireDefault(require("./sunshine-conversations-client.model/LocationRequest"));
var _Mailgun = _interopRequireDefault(require("./sunshine-conversations-client.model/Mailgun"));
var _MailgunAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/MailgunAllOf"));
var _MailgunUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/MailgunUpdate"));
var _MailgunUpdateAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/MailgunUpdateAllOf"));
var _MatchCriteria = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteria"));
var _MatchCriteriaBase = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteriaBase"));
var _MatchCriteriaMailgun = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteriaMailgun"));
var _MatchCriteriaMailgunAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteriaMailgunAllOf"));
var _MatchCriteriaMessagebird = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteriaMessagebird"));
var _MatchCriteriaMessagebirdAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteriaMessagebirdAllOf"));
var _MatchCriteriaTwilio = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteriaTwilio"));
var _MatchCriteriaTwilioAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteriaTwilioAllOf"));
var _MatchCriteriaWhatsapp = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteriaWhatsapp"));
var _MatchCriteriaWhatsappAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/MatchCriteriaWhatsappAllOf"));
var _Message = _interopRequireDefault(require("./sunshine-conversations-client.model/Message"));
var _MessageBirdUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/MessageBirdUpdate"));
var _MessageListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/MessageListResponse"));
var _MessageOverride = _interopRequireDefault(require("./sunshine-conversations-client.model/MessageOverride"));
var _MessageOverrideApple = _interopRequireDefault(require("./sunshine-conversations-client.model/MessageOverrideApple"));
var _MessageOverrideLine = _interopRequireDefault(require("./sunshine-conversations-client.model/MessageOverrideLine"));
var _MessageOverrideMessenger = _interopRequireDefault(require("./sunshine-conversations-client.model/MessageOverrideMessenger"));
var _MessageOverridePayload = _interopRequireDefault(require("./sunshine-conversations-client.model/MessageOverridePayload"));
var _MessageOverrideWhatsapp = _interopRequireDefault(require("./sunshine-conversations-client.model/MessageOverrideWhatsapp"));
var _MessagePost = _interopRequireDefault(require("./sunshine-conversations-client.model/MessagePost"));
var _MessagePostResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/MessagePostResponse"));
var _MessageWebhook = _interopRequireDefault(require("./sunshine-conversations-client.model/MessageWebhook"));
var _Messagebird = _interopRequireDefault(require("./sunshine-conversations-client.model/Messagebird"));
var _MessagebirdAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/MessagebirdAllOf"));
var _Messenger = _interopRequireDefault(require("./sunshine-conversations-client.model/Messenger"));
var _MessengerAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/MessengerAllOf"));
var _MessengerUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/MessengerUpdate"));
var _Meta = _interopRequireDefault(require("./sunshine-conversations-client.model/Meta"));
var _OfferControlBody = _interopRequireDefault(require("./sunshine-conversations-client.model/OfferControlBody"));
var _Page = _interopRequireDefault(require("./sunshine-conversations-client.model/Page"));
var _Participant = _interopRequireDefault(require("./sunshine-conversations-client.model/Participant"));
var _ParticipantJoinBody = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantJoinBody"));
var _ParticipantLeaveBody = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantLeaveBody"));
var _ParticipantLeaveBodyParticipantId = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantLeaveBodyParticipantId"));
var _ParticipantLeaveBodyUserExternalId = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantLeaveBodyUserExternalId"));
var _ParticipantLeaveBodyUserId = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantLeaveBodyUserId"));
var _ParticipantListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantListResponse"));
var _ParticipantResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantResponse"));
var _ParticipantSubSchema = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantSubSchema"));
var _ParticipantWithUserExternalId = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantWithUserExternalId"));
var _ParticipantWithUserId = _interopRequireDefault(require("./sunshine-conversations-client.model/ParticipantWithUserId"));
var _PassControlBody = _interopRequireDefault(require("./sunshine-conversations-client.model/PassControlBody"));
var _Postback = _interopRequireDefault(require("./sunshine-conversations-client.model/Postback"));
var _PostbackWebhook = _interopRequireDefault(require("./sunshine-conversations-client.model/PostbackWebhook"));
var _PrechatCapture = _interopRequireDefault(require("./sunshine-conversations-client.model/PrechatCapture"));
var _Profile = _interopRequireDefault(require("./sunshine-conversations-client.model/Profile"));
var _QuotedMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/QuotedMessage"));
var _QuotedMessageExternalMessageId = _interopRequireDefault(require("./sunshine-conversations-client.model/QuotedMessageExternalMessageId"));
var _QuotedMessageMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/QuotedMessageMessage"));
var _Referral = _interopRequireDefault(require("./sunshine-conversations-client.model/Referral"));
var _ReferralDetails = _interopRequireDefault(require("./sunshine-conversations-client.model/ReferralDetails"));
var _Reply = _interopRequireDefault(require("./sunshine-conversations-client.model/Reply"));
var _Source = _interopRequireDefault(require("./sunshine-conversations-client.model/Source"));
var _SourceWebhook = _interopRequireDefault(require("./sunshine-conversations-client.model/SourceWebhook"));
var _SourceWithCampaignWebhook = _interopRequireDefault(require("./sunshine-conversations-client.model/SourceWithCampaignWebhook"));
var _SourceWithCampaignWebhookAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/SourceWithCampaignWebhookAllOf"));
var _Status = _interopRequireDefault(require("./sunshine-conversations-client.model/Status"));
var _Switchboard = _interopRequireDefault(require("./sunshine-conversations-client.model/Switchboard"));
var _SwitchboardAcceptControl = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardAcceptControl"));
var _SwitchboardAcceptControlAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardAcceptControlAllOf"));
var _SwitchboardAcceptControlAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardAcceptControlAllOfPayload"));
var _SwitchboardAcceptControlFailure = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardAcceptControlFailure"));
var _SwitchboardAcceptControlFailureAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardAcceptControlFailureAllOf"));
var _SwitchboardAcceptControlFailureAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardAcceptControlFailureAllOfPayload"));
var _SwitchboardIntegration = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardIntegration"));
var _SwitchboardIntegrationCreateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardIntegrationCreateBody"));
var _SwitchboardIntegrationListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardIntegrationListResponse"));
var _SwitchboardIntegrationResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardIntegrationResponse"));
var _SwitchboardIntegrationUpdateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardIntegrationUpdateBody"));
var _SwitchboardIntegrationWebhook = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardIntegrationWebhook"));
var _SwitchboardListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardListResponse"));
var _SwitchboardOfferControl = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardOfferControl"));
var _SwitchboardOfferControlAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardOfferControlAllOf"));
var _SwitchboardOfferControlAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardOfferControlAllOfPayload"));
var _SwitchboardOfferControlFailure = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardOfferControlFailure"));
var _SwitchboardPassControl = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardPassControl"));
var _SwitchboardPassControlAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardPassControlAllOf"));
var _SwitchboardPassControlAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardPassControlAllOfPayload"));
var _SwitchboardPassControlFailure = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardPassControlFailure"));
var _SwitchboardResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardResponse"));
var _SwitchboardUpdateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/SwitchboardUpdateBody"));
var _Target = _interopRequireDefault(require("./sunshine-conversations-client.model/Target"));
var _Telegram = _interopRequireDefault(require("./sunshine-conversations-client.model/Telegram"));
var _TelegramAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/TelegramAllOf"));
var _TelegramUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/TelegramUpdate"));
var _TemplateMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/TemplateMessage"));
var _TextMessage = _interopRequireDefault(require("./sunshine-conversations-client.model/TextMessage"));
var _Twilio = _interopRequireDefault(require("./sunshine-conversations-client.model/Twilio"));
var _TwilioAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/TwilioAllOf"));
var _TwilioUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/TwilioUpdate"));
var _Twitter = _interopRequireDefault(require("./sunshine-conversations-client.model/Twitter"));
var _TwitterAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/TwitterAllOf"));
var _TwitterUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/TwitterUpdate"));
var _User = _interopRequireDefault(require("./sunshine-conversations-client.model/User"));
var _UserAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/UserAllOf"));
var _UserCreateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/UserCreateBody"));
var _UserMergeEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/UserMergeEvent"));
var _UserMergeEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/UserMergeEventAllOf"));
var _UserMergeEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/UserMergeEventAllOfPayload"));
var _UserMergeEventAllOfPayloadMergedClients = _interopRequireDefault(require("./sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedClients"));
var _UserMergeEventAllOfPayloadMergedConversations = _interopRequireDefault(require("./sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedConversations"));
var _UserMergeEventAllOfPayloadMergedUsers = _interopRequireDefault(require("./sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedUsers"));
var _UserResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/UserResponse"));
var _UserTruncated = _interopRequireDefault(require("./sunshine-conversations-client.model/UserTruncated"));
var _UserUpdateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/UserUpdateBody"));
var _UserUpdateEvent = _interopRequireDefault(require("./sunshine-conversations-client.model/UserUpdateEvent"));
var _UserUpdateEventAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/UserUpdateEventAllOf"));
var _UserUpdateEventAllOfPayload = _interopRequireDefault(require("./sunshine-conversations-client.model/UserUpdateEventAllOfPayload"));
var _Viber = _interopRequireDefault(require("./sunshine-conversations-client.model/Viber"));
var _ViberAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/ViberAllOf"));
var _ViberUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/ViberUpdate"));
var _Web = _interopRequireDefault(require("./sunshine-conversations-client.model/Web"));
var _WebAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/WebAllOf"));
var _WebUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/WebUpdate"));
var _WebUpdateAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/WebUpdateAllOf"));
var _Webhook = _interopRequireDefault(require("./sunshine-conversations-client.model/Webhook"));
var _WebhookBody = _interopRequireDefault(require("./sunshine-conversations-client.model/WebhookBody"));
var _WebhookCreateBody = _interopRequireDefault(require("./sunshine-conversations-client.model/WebhookCreateBody"));
var _WebhookListResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/WebhookListResponse"));
var _WebhookResponse = _interopRequireDefault(require("./sunshine-conversations-client.model/WebhookResponse"));
var _WebhookSubSchema = _interopRequireDefault(require("./sunshine-conversations-client.model/WebhookSubSchema"));
var _Webview = _interopRequireDefault(require("./sunshine-conversations-client.model/Webview"));
var _WhatsAppUpdate = _interopRequireDefault(require("./sunshine-conversations-client.model/WhatsAppUpdate"));
var _WhatsAppUpdateAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/WhatsAppUpdateAllOf"));
var _Whatsapp = _interopRequireDefault(require("./sunshine-conversations-client.model/Whatsapp"));
var _WhatsappAllOf = _interopRequireDefault(require("./sunshine-conversations-client.model/WhatsappAllOf"));
var _ActivitiesApi = _interopRequireDefault(require("./sunshine-conversations-client.api/ActivitiesApi"));
var _AppKeysApi = _interopRequireDefault(require("./sunshine-conversations-client.api/AppKeysApi"));
var _AppsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/AppsApi"));
var _AttachmentsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/AttachmentsApi"));
var _ClientsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/ClientsApi"));
var _ConversationsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/ConversationsApi"));
var _CustomIntegrationApiKeysApi = _interopRequireDefault(require("./sunshine-conversations-client.api/CustomIntegrationApiKeysApi"));
var _IntegrationsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/IntegrationsApi"));
var _MessagesApi = _interopRequireDefault(require("./sunshine-conversations-client.api/MessagesApi"));
var _OAuthEndpointsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/OAuthEndpointsApi"));
var _ParticipantsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/ParticipantsApi"));
var _SwitchboardActionsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/SwitchboardActionsApi"));
var _SwitchboardIntegrationsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/SwitchboardIntegrationsApi"));
var _SwitchboardsApi = _interopRequireDefault(require("./sunshine-conversations-client.api/SwitchboardsApi"));
var _UsersApi = _interopRequireDefault(require("./sunshine-conversations-client.api/UsersApi"));
var _WebhooksApi = _interopRequireDefault(require("./sunshine-conversations-client.api/WebhooksApi"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
